---
title: "R Notebook"
output: html_notebook
---

```{r}
source("functions.R")
source("packages.R")

```

```{r}
Table_of_genes <- read.csv("../Results/Genes_differentially_expressed.csv")

Table_of_methylations <- read.csv("../Results/CpGs_differentially_methylated.csv")
```

```{r}
# Cell type : "Mon", "Mac0", "Mac1", "Mac2", "Neu", "MK", "EP", "Ery", "FoeT", "nCD4", "tCD4", "aCD4", "naCD4", "nCD8", "tCD8", "nB", "tB"
pchic <- prepare_pchic()

PCHiC_GRange <- Create_pchic_Grange(pchic)

colnames(pchic) <- c("chr_bait", "start_bait", "end_bait", "ID_bait", "Name_bait", "chr_oe", "start_oe", "end_oe", "ID_oe", "Name_oe")
  pchic$IDbait <- paste(pchic$chr_bait, pchic$start_bait, sep = "_")
  pchic$IDoe <- paste(pchic$chr_oe, pchic$start_oe, sep = "_")

Blueprint_network <- read.csv("../BLUEPRINT_fragments_good.tsv", sep = "\t", stringsAsFactors = F) %>%
  dplyr::select(., "chr", "start", "end", "type", "ensembl", "gene_names", "intronic_regions", "type") %>% 
  separate_rows(., gene_names, sep = " ")

Blueprint_Granges <- GRanges(
  seqnames = Blueprint_network$chr,
  ranges = IRanges(start = Blueprint_network$start, end = Blueprint_network$end),
  Gene_name = Blueprint_network$gene_names,
  Promoter = Blueprint_network$type
)
```

### Overlapping between Chromatin fragment from PCHiC data and Blueprint data
```{r}
overlaps_Blueprint_pchic <- findOverlaps(PCHiC_GRange, Blueprint_Granges)
# partie metadata du granges en data frame, puis merge des deux metadata de phic_grange et BP_grange
match_hit_BP_Pchic <- data.frame(mcols(PCHiC_GRange[queryHits(overlaps_Blueprint_pchic), ]),
                                 data.frame(mcols(Blueprint_Granges[subjectHits(overlaps_Blueprint_pchic), ])),
                                 stringsAsFactors = T)

Nodes_in_BP_network <- unique(match_hit_BP_Pchic$ID)

match_hit_BP_promoter_pchic <- match_hit_BP_Pchic[match_hit_BP_Pchic$Promoter == "P",]
Nodes_in_BP_Promoter <- unique(match_hit_BP_promoter_pchic$ID)

Genes_BP <- unique(match_hit_BP_Pchic$Gene_name)
```

### Overlapping between differentially methylated CpGs and 3D Chromatin conformation
```{r}
CpGs_GRanges <- GRanges(
  seqnames = Table_of_methylations$CHR,
  ranges = IRanges(start = Table_of_methylations$MAPINFO, end = Table_of_methylations$MAPINFO+1),
  chr_cpg = Table_of_methylations$CHR,
  sens_of_methylation = Table_of_methylations$sens,
  CpG_position = Table_of_methylations$Relation_to_UCSC_CpG_Island
)

overlaps_CpGs_pchic <- findOverlaps(PCHiC_GRange, CpGs_GRanges)
match_hit_CpGs_Pchic <- data.frame(mcols(PCHiC_GRange[queryHits(overlaps_CpGs_pchic),]), data.frame(mcols(CpGs_GRanges[subjectHits(overlaps_CpGs_pchic),])))

Nodes_in_CpG <- unique(match_hit_CpGs_Pchic$ID)
```
### Overlapping between CpGs differentially methylated and Blueprint promoter reference
```{r}
CpGs_pchic_GRanges <- GRanges(
  seqnames = match_hit_CpGs_Pchic$chr_cpg,
  ranges = IRanges(start = match_hit_CpGs_Pchic$start_fragment, end = match_hit_CpGs_Pchic$end_fragment),
  chr_cpg = match_hit_CpGs_Pchic$chr_cpg,
  ID = match_hit_CpGs_Pchic$ID,
  sens_of_methylation = match_hit_CpGs_Pchic$sens_of_methylation,
  CpGs_position = match_hit_CpGs_Pchic$CpG_position
)

overlaps_CpGs_Blueprint <- findOverlaps(Blueprint_Granges, CpGs_pchic_GRanges)
match_hit_CpGs_Blueprint <- data.frame(mcols(Blueprint_Granges[queryHits(overlaps_CpGs_Blueprint),]),
                                       data.frame(mcols(CpGs_pchic_GRanges[subjectHits(overlaps_CpGs_Blueprint),])))

CpGs_nodes_promoter <- unique(match_hit_CpGs_Blueprint[match_hit_CpGs_Blueprint$Promoter == "P", "ID"])

CpGs_promoter <- unique(match_hit_CpGs_Blueprint[match_hit_CpGs_Blueprint$Promoter == "P",])

CpGs_not_promoter <- unique(match_hit_CpGs_Blueprint[match_hit_CpGs_Blueprint$Promoter == "O", "ID"])
```




### Analyses on overlapping CpGs differentially methylated and gene differencially expressed
```{r}
Genes_overlap_in_promoter <- CpGs_promoter[CpGs_promoter$Gene_name %in% Table_of_genes$hgnc_symbol,] %>% 
  merge(., Table_of_genes, by.x = "Gene_name", by.y = "hgnc_symbol")

write.csv(Genes_overlap_in_promoter, file = "../Results/Genes_overlap_in_promoter.csv", row.names = FALSE)
```

#### Summarize methylation regulation in promoter

```{r}
Gene_UP_Meth_UP <- Genes_overlap_in_promoter %>% dplyr::filter(., sens == "up_regulated" & sens_of_methylation == "hypermethylated")
Gene_UP_Meth_DOWN <- Genes_overlap_in_promoter %>% dplyr::filter(., sens == "up_regulated" & sens_of_methylation == "hypomethylated")
Gene_DOWN_Meth_UP <- Genes_overlap_in_promoter %>% dplyr::filter(., sens == "down_regulated" & sens_of_methylation == "hypermethylated")
Gene_DOWN_Meth_DOWN <- Genes_overlap_in_promoter %>% dplyr::filter(., sens == "down_regulated" & sens_of_methylation == "hypomethylated")

Sum_prom_regulation <- data.frame("Gene_UP_Meth_UP" = length(Gene_UP_Meth_UP$Gene_name), "Gene_UP_Meth_DOWN" = length(Gene_UP_Meth_DOWN$Gene_name), "Gene_DOWN_Meth_UP" = length(Gene_DOWN_Meth_UP$Gene_name), "Gene_DOWN_Meth_DOWN" = length(Gene_DOWN_Meth_DOWN$Gene_name))

write.csv(Sum_prom_regulation, file = "../Results/Summary_of_shortrange_regulation.csv", row.names = FALSE)
```





### Looking at Neighborhood of CpGs
```{r}
Neighbor_network <- unique(rbind(pchic[pchic$IDbait %in% match_hit_CpGs_Pchic$ID,],pchic[pchic$IDoe %in% match_hit_CpGs_Pchic$ID,]))[,c(1:3,11, 5:8,12, 10)]

colnames(Neighbor_network) <- rep(c("chr", "start", "end", "ID", "Gene_name"), 2)
Neighbor_nodes <- unique(rbind(Neighbor_network[,c(1:5)], Neighbor_network[,c(6:10)]))

Promoter_neighbors <- match_hit_BP_promoter_pchic[match_hit_BP_promoter_pchic$ID %in% Neighbor_nodes$ID & match_hit_BP_promoter_pchic$Promoter == "P", ]

Promoter_neighbors_genes <- unique(Promoter_neighbors$Gene_name)
```

#### Find neighbor genes that are differentially expressed
```{r}
Genes_DE_connected_to_CpGs_DM <- Promoter_neighbors[Promoter_neighbors$Gene_name %in% Table_of_genes$hgnc_symbol,]  %>%
  merge(., Table_of_genes, by.x = "Gene_name", by.y = "hgnc_symbol")
```

#### Analysis of the connections
```{r}
Genes_DE_connections <- unique(rbind(pchic[pchic$IDbait %in% Genes_DE_connected_to_CpGs_DM$ID,],pchic[pchic$IDoe %in% match_hit_CpGs_Pchic$ID,]))[,c(1:3,11, 5:8,12, 10)] %>%
  merge(., Genes_DE_connected_to_CpGs_DM, by.x = "IDbait", by.y = "ID") %>% 
  dplyr::select(., c("Gene_name", "logFC":"sens", "IDoe")) %>%
  merge(., match_hit_CpGs_Pchic, by.x = "IDoe", by.y = "ID") %>%
  dplyr::select(., c("Gene_name":"sens", "sens_of_methylation", "CpG_position"))

write.csv(Genes_DE_connections, file = "../Results/Genes_DE_connections.csv", row.names = FALSE)
```

#### Summarize methylation regulation in Oe
```{r}
Gene_UP_Oe_Meth_UP <- Genes_DE_connections %>% dplyr::filter(., sens == "up_regulated" & sens_of_methylation == "hypermethylated")
Gene_UP_Oe_Meth_DOWN <- Genes_DE_connections %>% dplyr::filter(., sens == "up_regulated" & sens_of_methylation == "hypomethylated")
Gene_DOWN_Oe_Meth_UP <- Genes_DE_connections %>% dplyr::filter(., sens == "down_regulated" & sens_of_methylation == "hypermethylated")
Gene_DOWN_Oe_Meth_DOWN <- Genes_DE_connections %>% dplyr::filter(., sens == "down_regulated" & sens_of_methylation == "hypomethylated")

Sum_Oe_regulation <- data.frame("Gene_UP_Oe_Meth_UP" = length(Gene_UP_Oe_Meth_UP$Gene_name), "Gene_UP_Oe_Meth_DOWN" = length(Gene_UP_Oe_Meth_DOWN$Gene_name), "Gene_DOWN_Oe_Meth_UP" = length(Gene_DOWN_Oe_Meth_UP$Gene_name), "Gene_DOWN_Oe_Meth_DOWN" = length(Gene_DOWN_Oe_Meth_DOWN$Gene_name))

write.csv(Sum_Oe_regulation, file = "../Results/Summary_of_longrange_regulation.csv", row.names = FALSE)

```