---
title: "R Notebook"
output: html_notebook
---

# INITIALISATION

```{r, include=FALSE}
source("functions.R")
BMIQ_list <- list()
Phenotypes <- list()
Results <- list()
Gene_ontology <- list()
source("Initialisation.R")
```

## DATA IMPORT

```{r}
load("../DATA/Methyl_DATA_Wang_Feng.RData")
load("../DATA/Wang_Feng_RNAseq.RData")
Phenotypes[["Koichi_cohort"]] <- read.csv("../DATA/Patient_phenotype_Koichi_cohort.csv")
```

# Methylation analysis

## Baseline Non_Responder vs Responder

```{r}
data_subset <- Phenotypes[["Koichi_cohort"]][ , c("Baseline_phenotype", "Baseline_Sample")]  

Patient_Baseline_with_treatment_annotation <- Phenotypes[["Koichi_cohort"]][complete.cases(data_subset), ]
rm(data_subset)
BMIQ_list[["Baseline_Methylation"]] <- dplyr::select(BMIQ_Wang_Feng, Patient_Baseline_with_treatment_annotation$Baseline_Sample)

Phenotypes[["Baseline_Methylation"]] <- data.frame("Sample" = Patient_Baseline_with_treatment_annotation$Baseline_Sample, "Phenotype" = Patient_Baseline_with_treatment_annotation$Baseline_phenotype)

Overlap_data[["EPIC_CpG_chromatin_enhancer_Koichi"]] <- Overlap_data[["EPIC_CpG_chromatin_enhancer"]] %>%
  dplyr::filter(., CpG %in% rownames(BMIQ_list[["Baseline_Methylation"]]))

tmp_data[["CpGs_EPIC_Non_Promoter_Koichi"]] <- Overlap_data[["EPIC_CpG_chromatin_enhancer_Koichi"]]$CpG %>%
  unique(.)
```

```{r}
ann_color_Meth <- list(
    Phenotype = c(Responder = "blue", Non_Responder = "red"))

png("../Results/Clustering_Baseline_dna_methylation.png", width = 360*(16/9), height = 360)
Make_heatmap(BMIQ_list[["Baseline_Methylation"]], Phenotype = Phenotypes[["Baseline_Methylation"]], "pearson", "Non_responder & Responder based on DNA methylation",  ann_color_Meth)
dev.off()
```


### Enhancer Koichi preparation

```{r}
BMIQ_list[["Koichi_Methylation_Non_promoter"]] <- BMIQ_Wang_Feng %>%
  dplyr::filter(., rownames(.) %in% tmp_data[["CpGs_EPIC_Non_Promoter_Koichi"]])

CpGs_per_fragment[["EPIC_Koichi"]] <- Overlap_data[["EPIC_CpG_chromatin_enhancer_Koichi"]] %>%
  dplyr::select(., c("CpG", "ID")) %>%
  split(., Overlap_data[["EPIC_CpG_chromatin_enhancer_Koichi"]]$ID)

part1 <- CpGs_per_fragment[["EPIC_Koichi"]][c(1:20000)]

part1 <- lapply(part1, function(fragment){
  tmp <- BMIQ_list[["Koichi_Methylation_Non_promoter"]][fragment$CpG,] %>%
    as.matrix(.)
  if (length(tmp[,1])>1){
    tmp <- t(colMeans2(tmp))
    colnames(tmp) <- colnames(BMIQ_list[["Koichi_Methylation_Non_promoter"]])
  }
  as.data.frame(tmp)
})
system("say part one finished")
part2 <- CpGs_per_fragment[["EPIC_Koichi"]][c(20001:40000)]

part2 <- lapply(part2, function(fragment){
  tmp <- BMIQ_list[["Koichi_Methylation_Non_promoter"]][fragment$CpG,] %>%
    as.matrix(.)
  if (length(tmp[,1])>1){
    tmp <- t(colMeans2(tmp))
    colnames(tmp) <- colnames(BMIQ_list[["Koichi_Methylation_Non_promoter"]])
  }
  as.data.frame(tmp)
})
system("say part two finished")
part3 <- CpGs_per_fragment[["EPIC_Koichi"]][c(40001:60000)]

part3 <- lapply(part3, function(fragment){
  tmp <- BMIQ_list[["Koichi_Methylation_Non_promoter"]][fragment$CpG,] %>%
    as.matrix(.)
  if (length(tmp[,1])>1){
    tmp <- t(colMeans2(tmp))
    colnames(tmp) <- colnames(BMIQ_list[["Koichi_Methylation_Non_promoter"]])
  }
  as.data.frame(tmp)
})

system("say part three finished")

part4 <- CpGs_per_fragment[["EPIC_Koichi"]][c(60001:68371)]

part4 <- lapply(part4, function(fragment){
  tmp <- BMIQ_list[["Koichi_Methylation_Non_promoter"]][fragment$CpG,] %>%
    as.matrix(.)
  if (length(tmp[,1])>1){
    tmp <- t(colMeans2(tmp))
    colnames(tmp) <- colnames(BMIQ_list[["Koichi_Methylation_Non_promoter"]])
  }
  as.data.frame(tmp)
})

system("say finished finished finished")

CpGs_per_fragment[["Means_EPIC_Koichi"]] <- c(part1, part2, part3, part4)

saveRDS(CpGs_per_fragment[["Means_EPIC_Koichi"]], "../DATA/Means_Cpgs_Koichi_per_fragment.RDS")
```

### Promoter

```{r}
Results[["Basline_methylation_promoter"]] <- T_test_on_methylation_promoter(BMIQ = BMIQ_list[["Baseline_Methylation"]], 
                                                                            Phenotype = Phenotypes[["Baseline_Methylation"]],
                                                                            match_hit_CpGs_Blueprint_promoter = Overlap_data[["EPIC_Blueprint_promoter"]],
                                                                            comparison = c("Non_Responder", "Responder"))

```

#### Gene ontology

```{r}
png(filename = "../Results/Gene_ontology_promoter.png", width = 480*(16/9), height = 480)
emapplot(Results[["Basline_methylation_promoter"]]$GO_hyper, color = "qvalue", size = "Count")
dev.off()
```

```{r}
emapplot(Results[["Basline_methylation_promoter"]]$GO_hypo, color = "qvalue", size = "Count")
```


### Enhancer

```{r}
Samples <- colnames(CpGs_per_fragment[["Means_EPIC_Koichi"]][["1_172895418"]]) 

A <- sapply(Samples, function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Non_Responder")
  x %in% tmp$Baseline_Sample
}) %>% as.vector(.)
B <- sapply(Samples, function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Responder")
  x %in% tmp$Baseline_Sample
}) %>% as.vector(.)

comparison <- list()
comparison[[1]] <- A
comparison[[2]] <- B

Results[["Baseline_enhancer_analysis"]] <- T_test_on_methylation_enhancer(Mean_CpGs_per_fragment = CpGs_per_fragment[["Means_EPIC_Koichi"]], 
                               comparisons = comparison, 
                               Fragment_connected_per_gene = Overlap_data[["Fragment_connected_per_gene"]])

Results[["Baseline_enhancer_analysis"]][["Genes_list_of_Genes_with_hyper_enhancers_annoted"]] %>% write.csv(., "../Results/Tables report/Baseline_enhancer_analysis_genes_hyper.csv")
```

#### Gene ontology
```{r}
png(filename = "../Results/Gene_ontology_enhancers.png", width = 480*(16/9), height = 480)
dotplot(Results[["Baseline_enhancer_analysis"]]$GO, showCategory = 15)
dev.off()

dotplot(Results[["Baseline_enhancer_analysis"]]$GO, showCategory = 15)
emapplot(Results[["Baseline_enhancer_analysis"]]$GO, color = "qvalue", size = "Count")

png(filename = "../Results/Gene_ontology_enhancers_map.png", width = 360*(16/9), height = 360)
emapplot(Results[["Baseline_enhancer_analysis"]]$GO, color = "qvalue", size = "Count")
dev.off()

```



## After treatment Non_responder vs Responder


```{r}
data_subset <- Phenotypes[["Koichi_cohort"]][ , c("Baseline_phenotype", "Post_treatment_sample")]  

Patient_post_treatment_with_treatment_annotation <- Phenotypes[["Koichi_cohort"]][complete.cases(data_subset), ]
rm(data_subset)
BMIQ_list[["Post_Methylation"]] <- dplyr::select(BMIQ_Wang_Feng, Patient_post_treatment_with_treatment_annotation$Post_treatment_sample)

Phenotypes[["Post_Methylation"]] <- data.frame("Sample" = Patient_Baseline_with_treatment_annotation$Post_treatment_sample, "Phenotype" = Patient_Baseline_with_treatment_annotation$Baseline_phenotype)
```

### Promoter

```{r}
Results[["Post_methylation_promoter"]] <- T_test_on_methylation_promoter(BMIQ = BMIQ_list[["Post_Methylation"]], 
                                                                            Phenotype = Phenotypes[["Post_Methylation"]],
                                                                            match_hit_CpGs_Blueprint_promoter = Overlap_data[["EPIC_Blueprint_promoter"]],
                                                                            comparison = c("Non_Responder", "Responder"))
```

### Enhancer

```{r}
A <- sapply(Samples, function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Non_Responder")
  x %in% tmp$Post_treatment_sample
}) %>% as.vector(.)
B <- sapply(Samples, function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Responder")
  x %in% tmp$Post_treatment_sample
}) %>% as.vector(.)

comparison <- list()
comparison[[1]] <- A
comparison[[2]] <- B

Results[["After_treatment_enhancer_analysis"]] <- T_test_on_methylation_enhancer(Mean_CpGs_per_fragment = CpGs_per_fragment[["Means_EPIC_Koichi"]], 
                               comparisons = comparison, 
                               Fragment_connected_per_gene = Overlap_data[["Fragment_connected_per_gene"]])

Results[["After_treatment_enhancer_analysis"]][["Genes_list_of_Genes_with_hyper_enhancers_annoted"]] %>% write.csv(., "../Results/Tables report/After_treatment_enhancer_analysis_genes_hyper.csv")
```

# Gene expression


```{r}
A <- sapply(colnames(RNAseq_Wang_Feng), function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Non_Responder")
  x %in% tmp$Baseline_RNAseq_data
})

B <- sapply(colnames(RNAseq_Wang_Feng), function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Responder")
  x %in% tmp$Baseline_RNAseq_data
})

RNAseq_Wang_Feng_Baseline <- RNAseq_Wang_Feng %>%
  .[,A | B]

C <- sapply(colnames(RNAseq_Wang_Feng_Baseline), function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Non_Responder")
  x %in% tmp$Baseline_RNAseq_data
})

D <- sapply(colnames(RNAseq_Wang_Feng_Baseline), function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Responder")
  x %in% tmp$Baseline_RNAseq_data
})

tmp <- data.frame("Non_responder" = C, "Responder" = D)
tmp$Phenotype <- ifelse(C, "Non_responder", ifelse(D, "Responder", "No_data"))

ann_color_RNA <- list(
    Phenotype = c(Responder = "blue", Non_responder = "red", No_data = "grey"))



png("../Results/Heatmap_RNAseq_Baseline.png")
Make_heatmap(RNAseq_Wang_Feng_Baseline, tmp, "pearson", "Non_responder & Responder", ann_color_RNA)
dev.off()

heatmap_Baseline <- Make_heatmap(RNAseq_Wang_Feng_Baseline, tmp, "pearson", "Non_responder & Responder", ann_color_RNA)

# colnames(RNAseq_Wang_Feng)[heatmap_Baseline$tree_col[["order"]]][c(1:15)]
# 
# batch = c(rep("A", 15), rep("B", 51-15))
# 
# combat_edata1 = ComBat(dat=RNAseq_Wang_Feng, batch=batch, mod=NULL, par.prior=TRUE, prior.plots=FALSE)
# 
# Make_heatmap(combat_edata1, tmp, "pearson", "Non_responder & Responder", ann_color_RNA)

RNAseq_Wang_Feng_Baseline <- voom(RNAseq_Wang_Feng_Baseline)

RNAseq_Wang_Feng_Baseline_analysis <- Differential_analysis(tmp$Phenotype, RNAseq_Wang_Feng_Baseline)


RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$logFC <- ifelse(RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$logFC > 1, log(RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$logFC), ifelse(RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$logFC < -1 , -log(-RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$logFC), 0))

# RNAseq_Wang_Feng_Baseline_analysis_batch1_corrected <- Differential_analysis(tmp$Phenotype, combat_edata1)

Genes_diff_expressed <- dplyr::filter(RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]], abs(logFC) > 0.75 & P.Value < 0.05) %>% 
  .$ID
```


### Clustering 

```{r}
colnames(RNAseq_Baselin_rownames) <- substring(colnames(RNAseq_Baselin_rownames), 2,11)
corr_RNA <- rcorr(as.matrix(RNAseq_Baselin_rownames), type = "pearson")$r

annotation_heatmap <- data.frame("Phenotype" = tmp$Phenotype)
rownames(annotation_heatmap) <- rownames(tmp)

ann_color_RNA <- list(
    Phenotype = c(Responder = "blue", Non_responder = "red")
    )

png("../Results/Corr_RNAseq.png",  width = 360*(4/3), height = 360)
pheatmap(corr_RNA, annotation_row = annotation_heatmap, annotation_col = annotation_heatmap, annotation_colors = ann_color_RNA, show_rownames = FALSE, show_colnames = FALSE, annotation_names_row = FALSE)
dev.off()
pheatmap(corr_RNA, annotation_row = annotation_heatmap, annotation_col = annotation_heatmap, annotation_colors = ann_color_RNA, show_rownames = FALSE, show_colnames = FALSE, annotation_names_row = FALSE)
```




```{r}
RNA_DM_enhancer_promoter <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% c(Genes_DM_enhancer,Genes_DM_promoter)),]
Make_heatmap(RNA_DM_enhancer_promoter, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DM (enhancer/promoter)", ann_color_RNA)

png("../Results/Clustering_Baseline_based_on_gene_with_promoter_or_enhancer_methylated.png", width = 360*(16/9), height = 360)
Make_heatmap(RNA_DM_enhancer_promoter, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DM (enhancer/promoter)", ann_color_RNA)
dev.off()
```

```{r}
RNA_DM_enhancer_promoter <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% c(Genes_DM_enhancer,Genes_DM_promoter)),]
colnames(RNA_DM_enhancer_promoter) <- substring(colnames(RNA_DM_enhancer_promoter), 2,11)

pheatmap(as.matrix(RNA_DM_enhancer_promoter),
main="AML Patient Clustering based on Gene expression Data filtered for DM (promoter/enhancer)",
show_rownames = FALSE,
annotation_col = annotation_heatmap,
annotation_colors = ann_color_RNA,
clustering_distance_rows="euclidean",
scale = "column")
```

```{r}
RNA_DM_enhancer_Down <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% Genes_DM_enhancer_Down),]
colnames(RNA_DM_enhancer_Down) <- substring(colnames(RNA_DM_enhancer_Down), 2,11)
Make_heatmap(RNA_DM_enhancer_Down, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DM (enhancer) & Down", ann_color_RNA)

png("../Results/Clustering_Baseline_based_on_gene_Down_with_enhancer_methylated.png", width = 360*(16/9), height = 360)
Make_heatmap(RNA_DM_enhancer_Down, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DM (enhancer) & Down", ann_color_RNA)
dev.off()
```

```{r}
RNA_DM_enhancer_DE <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% Genes_DM_enhancer_DE),]
colnames(RNA_DM_enhancer_DE) <- substring(colnames(RNA_DM_enhancer_DE), 2,11)
Make_heatmap(RNA_DM_enhancer_DE, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DM (enhancer) & DE", ann_color_RNA)

png("../Results/Clustering_Baseline_based_on_gene_with_enhancer_methylated.png", width = 360*(16/9), height = 360)
Make_heatmap(RNA_DM_enhancer_DE, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DM (enhancer) & DE", ann_color_RNA)
dev.off()
```

```{r}
RNA_DM_enhancer_Down <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% Genes_DM_enhancer),]
colnames(RNA_DM_enhancer_DE) <- substring(colnames(RNA_DM_enhancer_DE), 2,11)

Make_heatmap(RNA_DM_enhancer_DE, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DM (enhancer) & DE", ann_color_RNA)
```


```{r}
RNA_DM_promoter <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% Genes_DM_promoter),] 
colnames(RNA_DM_promoter) <- substring(colnames(RNA_DM_promoter), 2,11)
outliers <- which(colnames(RNA_DM_promoter) %ni% c("2292915_BL", "2163337_BL"))
Phenotype_no_outliers <- which(rownames(tmp) %ni% c("2292915_BL", "2163337_BL"))

Phenotype_no_outliers <- tmp[Phenotype_no_outliers,]
RNA_DM_promoter <- dplyr::select(RNA_DM_promoter, outliers)

png("../Results/Clustering_Baseline_based_on_gene_with_promoter_methylated.png", width = 360*(16/9), height = 360)
Make_heatmap(RNA_DM_promoter, Phenotype = Phenotype_no_outliers, "pearson", "Non_responder & Responder based on Genes DM (promoter)", ann_color_RNA)
dev.off()
```

```{r}
pheatmap(as.matrix(RNA_DM_promoter),
main="AML Patient Clustering based on Gene expression Data filtered for DM (promoter)",
show_rownames = FALSE,
annotation_col = annotation_heatmap,
annotation_colors = ann_color_RNA,
clustering_distance_rows="euclidean")
```


```{r}
RNA_DM_enhancer <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% Genes_DM_enhancer),]
colnames(RNA_DM_enhancer) <- substring(colnames(RNA_DM_enhancer), 2,11)

Make_heatmap(RNA_DM_enhancer, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DM (enhancer)", ann_color_RNA)
```

```{r}
pheatmap(as.matrix(RNA_DM_enhancer),
main="AML Patient Clustering based on Gene expression Data filtered for DM (enhancer)",
show_rownames = FALSE,
annotation_col = annotation_heatmap,
annotation_colors = ann_color_RNA,
clustering_distance_rows="euclidean",
scale = "column")
```



```{r}
RNA_DE <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% c(Genes_diff_expressed)),]
colnames(RNA_DE) <- substring(colnames(RNA_DE), 2,11)

Make_heatmap(voom(RNA_DE)$E, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes DE", ann_color_RNA)
```

```{r}
pheatmap(as.matrix(voom(RNA_DE)$E),
main="AML Patient Clustering based on Gene expression Data filtered for DE genes",
show_rownames = FALSE,
annotation_col = annotation_heatmap,
annotation_colors = ann_color_RNA,
clustering_distance_rows="euclidean")
```

```{r}
Genes_target <- read.csv("../Results/regulons_msviper.csv(2) default node.csv") %>% .$name %>% str_remove_all("\"")

target_RNA <- RNAseq_Baselin_rownames %>% .[which(rownames(RNAseq_Baselin_rownames) %in% Genes_target),]

target_RNA <- voom(target_RNA)

pheatmap(as.matrix(target_RNA$E),
  main="AML Patient Clustering based on Genes DE connected to TF DA",
  show_rownames = TRUE,
  annotation_col = annotation_heatmap,
  annotation_colors = ann_color_RNA,
  clustering_distance_rows="euclidean")
```



```{r}
Genes_methylation <- Results[["Basline_methylation_promoter"]]$Genes_methylation_analysis
colnames(Genes_methylation) <- c("Gene_name", "Methylation_p.value", "Mean_NR_methylation", "Mean_R_methylation", "delta_methylation")

Genes_expression <- RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]
Genes_expression$ID <- Genes_expression$ID %>% str_split(., "[|]") %>% lapply(., function(x){
  x[1]
}) %>%
  unlist(.)

# Genes_expression_batch_corrected <- RNAseq_Wang_Feng_Baseline_analysis_batch1_corrected[["Non_responder-Responder"]]
# Genes_expression_batch_corrected$ID <- Genes_expression_batch_corrected$ID %>% str_split(., "[|]") %>% lapply(., function(x){
#   x[1]
# }) %>%
#   unlist(.)

Gene_methyl_expression_data <- merge(Genes_expression, Genes_methylation, by.x = "ID", by.y = "Gene_name", all = TRUE)
Gene_methyl_expression_data_both_annotations <- merge(Genes_expression, Genes_methylation, by.x = "ID", by.y = "Gene_name")

Genes_hypermethylated_NR_vs_R <- Gene_methyl_expression_data_both_annotations %>%
  dplyr::filter(., delta_methylation > 0.1 & Methylation_p.value < 0.1)

Genes_hypomethylated_NR_vs_R <- Gene_methyl_expression_data_both_annotations %>%
  dplyr::filter(., delta_methylation < -0.1 & Methylation_p.value < 0.1)

Genes_Hypermethylated_suppressed <- Gene_methyl_expression_data_both_annotations %>%
  dplyr::filter(., delta_methylation > 0.1 & logFC < -0.75 & P.Value < 0.1 & Methylation_p.value < 0.1)

Genes_Hypomethylated_upregulated <-  Gene_methyl_expression_data_both_annotations %>%
  dplyr::filter(., delta_methylation < -0.1 & logFC > 0.75 & P.Value < 0.1 & Methylation_p.value < 0.1)

Genes_Hypermethylated_upregulated <- Gene_methyl_expression_data_both_annotations %>%
  dplyr::filter(., delta_methylation > 0.1 & logFC > 0.75 & P.Value < 0.1 & Methylation_p.value < 0.1)

Genes_Hypomethylated_suppressed <-  Gene_methyl_expression_data_both_annotations %>%
  dplyr::filter(., delta_methylation < -0.1 & logFC < -0.75 & P.Value < 0.1 & Methylation_p.value < 0.1)


###############

# Gene_methyl_expression_data_batch_corrected <- merge(Genes_expression_batch_corrected, Genes_methylation, by.x = "ID", by.y = "Gene_name", all = TRUE)
# 
# Genes_hypermethylated_NR_vs_R_batch_corrected <- Gene_methyl_expression_data_batch_corrected %>%
#   dplyr::filter(., delta_methylation > 0.1 & Methylation_p.value < 0.1)
# 
# Genes_hypomethylated_NR_vs_R_batch_corrected <- Gene_methyl_expression_data_batch_corrected %>%
#   dplyr::filter(., delta_methylation < -0.1 & Methylation_p.value < 0.1)
# 
# Genes_Hypermethylated_suppressed_batch_corrected <- Gene_methyl_expression_data_batch_corrected %>%
#   dplyr::filter(., delta_methylation > 0.1 & logFC < -0.75 & P.Value < 0.1 & Methylation_p.value < 0.1)
# 
# Genes_Hypomethylated_upregulated_batch_corrected <-  Gene_methyl_expression_data_batch_corrected %>%
#   dplyr::filter(., delta_methylation < -0.1 & logFC > 0.75 & P.Value < 0.1 & Methylation_p.value < 0.1)
# 
# Genes_Hypermethylated_upregulated_batch_corrected <- Gene_methyl_expression_data_batch_corrected %>%
#   dplyr::filter(., delta_methylation > 0.1 & logFC > 0.75 & P.Value < 0.1 & Methylation_p.value < 0.1)
# 
# Genes_Hypomethylated_suppressed_batch_corrected <-  Gene_methyl_expression_data_batch_corrected %>%
#   dplyr::filter(., delta_methylation < -0.1 & logFC < -0.75 & P.Value < 0.1 & Methylation_p.value < 0.1)
```

```{r}
Number_Genes_Hypermethylated_no_changes <- length(Genes_hypermethylated_NR_vs_R$ID) - length(Genes_Hypermethylated_upregulated$ID) - length(Genes_Hypermethylated_suppressed$ID)

Number_Genes_Hypomethylated_no_changes <- length(Genes_hypomethylated_NR_vs_R$ID) - length(Genes_Hypomethylated_upregulated$ID) - length(Genes_Hypomethylated_suppressed$ID)

Numbers <- c(length(Genes_Hypermethylated_upregulated$ID), 
             Number_Genes_Hypermethylated_no_changes, 
             length(Genes_Hypermethylated_suppressed$ID), 
             length(Genes_Hypomethylated_upregulated$ID), 
             Number_Genes_Hypomethylated_no_changes, 
             length(Genes_Hypomethylated_suppressed$ID))


Barplot_data <- data.frame(Gene_expression=rep(c("up_regulated", "no_changes", "down_regulated"), 2),
                           Methylation=rep(c("Hypermethylated", "Hypomethylated"), each = 3),
                           Number_of_genes = Numbers)

ggplot(data=Barplot_data, aes(x=Methylation, y=Number_of_genes, fill=Gene_expression)) +
  geom_bar(stat="identity")
ggsave("~/correlation_methylation_expression.png")
```

```{r}
Genes_up <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., logFC > 0.75 & P.Value < 0.05) %>%
  .$ID

Genes_hyper <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., Methylation_p.value < 0.05 & delta_methylation > 0.1) %>%
  .$ID

Genes_hypo <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., Methylation_p.value < 0.05 & delta_methylation < -0.1) %>%
  .$ID

Genes_down <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., logFC < -0.75 & P.Value < 0.05) %>%
  .$ID

Venn_promoter_diagramm <- list(
  "Hyper" = Genes_hyper,
  "Down" = Genes_down,
  "UP" = Genes_up,
  "Hypo" = Genes_hypo
)
library(ggvenn)
ggvenn(
  Venn_promoter_diagramm, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"))
#ggsave("~/Venn_expression_methylation_promoter.png")
```

```{r}
Genes_up <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., logFC > 0.75 & P.Value < 0.05) %>%
  .$ID

Genes_DM <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., Methylation_p.value < 0.05 & abs(delta_methylation) > 0.1) %>%
  .$ID

Genes_down <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., logFC < -0.75 & P.Value < 0.05) %>%
  .$ID

Venn_promoter_diagramm <- list(
  "DM" = Genes_DM,
  "Down" = Genes_down,
  "UP" = Genes_up
)
library(ggvenn)
ggvenn(
  Venn_promoter_diagramm, 
  fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF"))
#ggsave("~/Venn_expression_Dmethylation_promoter.png")

table_fisher_test_promoter <- data.frame("Methylated_promoter" = c(32, 324), "unmethylated_promoter" = c(897, 15215))
rownames(table_fisher_test_promoter) <- c("DE", "no DE")

fisher.test(table_fisher_test_promoter)


```


### Enhancer

```{r}
tmp_names <- names(Results[["Baseline_enhancer_analysis"]][["Enhancer_summary"]])
  
Gene_enhancer_methylation_analysis <- rbindlist(Results[["Baseline_enhancer_analysis"]][["Enhancer_summary"]]) %>% as.data.frame(.)
rownames(Gene_enhancer_methylation_analysis) <- tmp_names

Gene_methyl_enhancer_expression_data <- merge(Genes_expression, Gene_enhancer_methylation_analysis, by.x = "ID", by.y = 0)

Gene_methyl_enhancer_expression_data$logFC <- ifelse(Gene_methyl_enhancer_expression_data$logFC > 1, log10(Gene_methyl_enhancer_expression_data$logFC),  ifelse(Gene_methyl_enhancer_expression_data$logFC < -1, -log10(-Gene_methyl_enhancer_expression_data$logFC), Gene_methyl_enhancer_expression_data$logFC))

Gene_methyl_enhancer_expression_data$nb_enhancer_changed <- Gene_methyl_enhancer_expression_data$nb_enhancer_hyper + Gene_methyl_enhancer_expression_data$nb_enhancer_hypo
```


```{r}
nb_enhancer_treshold <- 0

Number_Genes_Hypermethylated_no_changes <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & (P.Value > 0.05 | abs(logFC) < 0.75)) %>%
  .$ID %>%
  length()
  
Number_Genes_Hypermethylated_enh_suppressed <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & P.Value < 0.05 & logFC < -0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypermethylated_enh_surrexpressed <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & P.Value < 0.05 & logFC > 0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypomethylated_no_changes <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & (P.Value > 0.05 | abs(logFC)) < 0.75) %>%
  .$ID %>%
  length()
  
Number_Genes_Hypomethylated_enh_suppressed <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & P.Value < 0.05 & logFC < -0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypomethylated_enh_surrexpressed <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & P.Value < 0.05 & logFC > 0.75) %>%
  .$ID %>%
  length()

Numbers <- c(Number_Genes_Hypermethylated_enh_suppressed, 
             Number_Genes_Hypermethylated_no_changes, 
             Number_Genes_Hypermethylated_enh_surrexpressed, 
             Number_Genes_Hypomethylated_enh_suppressed, 
             Number_Genes_Hypomethylated_no_changes, 
             Number_Genes_Hypomethylated_enh_surrexpressed)


Barplot_data <- data.frame(Gene_expression=rep(c("down_regulated", "no_changes", "up_regulated"), 2),
                           Methylation=rep(c("Hypermethylated", "Hypomethylated"), each = 3),
                           Number_of_genes = Numbers)

ggplot(data=Barplot_data, aes(x=Methylation, y=Number_of_genes, fill=Gene_expression)) +
  geom_bar(stat="identity")
ggsave("~/correlation_methylation_enhancer_expression.png")
```

```{r}
nb_enhancer_treshold <- 0

Genes_up <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., logFC > 0.75 & P.Value < 0.05) %>%
  .$ID

Genes_hyper <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold) %>%
  .$ID

Genes_hypo <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold) %>%
  .$ID

Genes_down <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., logFC < -0.75 & P.Value < 0.05) %>%
  .$ID

Venn_promoter_diagramm <- list(
  "Hyper" = Genes_hyper,
  "Down" = Genes_down,
  "UP" = Genes_up,
  "Hypo" = Genes_hypo
)
library(ggvenn)
ggvenn(
  Venn_promoter_diagramm, 
  fill_color = c("#0073C2FF", "#EFC000FF", "#868686FF", "#CD534CFF"))
ggsave("~/Venn_expression_methylation_enhancer.png")
```

```{r}
nb_enhancer_treshold <- 0

Genes_up <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., logFC > 0.75 & P.Value < 0.05) %>%
  .$ID

Genes_DM <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold | nb_enhancer_hypo > nb_enhancer_treshold) %>%
  .$ID

Genes_down <- Gene_methyl_enhancer_expression_data %>% 
  dplyr::filter(., logFC < -0.75 & P.Value < 0.05) %>%
  .$ID

Venn_promoter_diagramm <- list(
  "DM" = Genes_DM,
  "Down" = Genes_down,
  "UP" = Genes_up
)

genes_DM_DE <- c(Genes_down, Genes_up) %>% 
  intersect(., Genes_DM) %>%
  length()

genes_no_DM_DE <- c(Genes_down, Genes_up) %>%
  length()
genes_no_DM_DE <- genes_no_DM_DE - genes_DM_DE

genes_DM_noDE <- Genes_DM %>%
  length()
genes_DM_noDE <- genes_DM_noDE - genes_DM_DE

genes_no_DM_noDE <- Gene_methyl_enhancer_expression_data$ID %>%
  unique(.) %>%
  length()
genes_no_DM_noDE <- genes_no_DM_noDE - genes_DM_noDE - genes_no_DM_DE - genes_DM_DE

library(ggvenn)
ggvenn(
  Venn_promoter_diagramm, 
  fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF"))
ggsave("~/Venn_expression_Dmethylation_enhancer.png")



table_fisher_test <- data.frame("Methylated_enhancer" = c(624, 5941), "Unmethylated_enhancer" = c(650, 20512))
rownames(table_fisher_test) <- c("DE", "no DE")

fisher.test(table_fisher_test)

```

```{r}
Genes_up_DM <- Gene_methyl_enhancer_expression_data %>% dplyr::filter(., logFC > 0.75 & P.Value < 0.05 & (nb_enhancer_hyper > nb_enhancer_treshold | nb_enhancer_hypo > nb_enhancer_treshold))
Genes_down_DM <- Gene_methyl_enhancer_expression_data %>% dplyr::filter(., logFC < -0.75 & P.Value < 0.05 & (nb_enhancer_hyper > nb_enhancer_treshold | nb_enhancer_hypo > nb_enhancer_treshold))

Gene_enhancer_DM_expression <- rbind(Genes_up_DM, Genes_down_DM)

write.csv(Gene_enhancer_DM_expression, "../Results/Tables report/Gene lists/Gene_enhancer_DM_expression.csv", row.names = FALSE)
```


```{r}
Gene_methyl_enhancer_expression_data_target_gene <- dplyr::filter(Gene_methyl_enhancer_expression_data, ID %in% c(Genes_Hypermethylated_suppressed$ID, Genes_Hypomethylated_suppressed$ID))

nb_enhancer_treshold <- 1

Number_Genes_Hypermethylated_no_changes <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & (P.Value > 0.05 | abs(logFC) < 0.75)) %>%
  .$ID %>%
  length()
  
Number_Genes_Hypermethylated_enh_suppressed <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & P.Value < 0.05 & logFC < -0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypermethylated_enh_surrexpressed <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & P.Value < 0.05 & logFC > 0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypomethylated_no_changes <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & (P.Value > 0.05 | abs(logFC)) < 0.75) %>%
  .$ID %>%
  length()
  
Number_Genes_Hypomethylated_enh_suppressed <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & P.Value < 0.05 & logFC < -0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypomethylated_enh_surrexpressed <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & P.Value < 0.05 & logFC > 0.75) %>%
  .$ID %>%
  length()

Numbers <- c(Number_Genes_Hypermethylated_enh_suppressed, 
             Number_Genes_Hypermethylated_no_changes, 
             Number_Genes_Hypermethylated_enh_surrexpressed, 
             Number_Genes_Hypomethylated_enh_suppressed, 
             Number_Genes_Hypomethylated_no_changes, 
             Number_Genes_Hypomethylated_enh_surrexpressed)


Barplot_data <- data.frame(Gene_expression=rep(c("down_regulated", "no_changes", "up_regulated"), 2),
                           Methylation=rep(c("Hypermethylated", "Hypomethylated"), each = 3),
                           Number_of_genes = Numbers)

ggplot(data=Barplot_data, aes(x=Methylation, y=Number_of_genes, fill=Gene_expression)) +
  geom_bar(stat="identity")
ggsave("~/correlation_methylation_enhancer_promoter_expression.png")
```

```{r}
Genes_hypermethylated_analysis <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., Methylation_p.value < 0.1 & delta_methylation > 0.1)

enhancedvolcano_data1 <- data.frame(FC = Genes_hypermethylated_analysis$logFC,
                                    p.value = Genes_hypermethylated_analysis$P.Value,
                                    gene = Genes_hypermethylated_analysis$ID)

enhancedvolcano_data1$logFC <- ifelse(enhancedvolcano_data1$FC > 0, log(enhancedvolcano_data1$FC), -log(-enhancedvolcano_data1$FC))

EnhancedVolcano(toptable = enhancedvolcano_data1,
                lab = enhancedvolcano_data1$gene,
                x = "logFC",
                y = "p.value",
                FCcutoff = 0.75,
                pCutoff = 0.1,
                ylim = c(0, 5),
                xlim = c(-15,15)
                )
ggsave("../Results/correlation_hypermetylation_expression.png")

Genes_hypermethylated_analysis %>% 
  dplyr::filter(., logFC < -0.75 & P.Value < 0.1) %>%
  .$ID %>%
  write.csv(., "../Results/Tables report/Gene lists/Genes_hypermeth_suppressed.csv", row.names = FALSE)

Genes_hypermethylated_analysis %>% 
  dplyr::filter(., logFC > 0.75 & P.Value < 0.1) %>%
  .$ID %>%
  write.csv(., "../Results/Tables report/Gene lists/Genes_hypermeth_surressed.csv", row.names = FALSE)
```


```{r}
Genes_hypomethylated_analysis <- Gene_methyl_expression_data_both_annotations %>% 
  dplyr::filter(., Methylation_p.value < 0.05 & delta_methylation < -0.1)

enhancedvolcano_data2 <- data.frame(FC = Genes_hypomethylated_analysis$logFC,
                                    p.value = Genes_hypomethylated_analysis$P.Value,
                                    gene = Genes_hypomethylated_analysis$ID)

enhancedvolcano_data2$logFC <- ifelse(enhancedvolcano_data2$FC > 0, log(enhancedvolcano_data2$FC), -log(-enhancedvolcano_data2$FC))

EnhancedVolcano(toptable = enhancedvolcano_data2,
                lab = enhancedvolcano_data2$gene,
                x = "logFC",
                y = "p.value",
                FCcutoff = 0.75,
                pCutoff = 0.05,
                ylim = c(0, 5),
                xlim = c(-15,15)
                )
ggsave("~/correlation_hypomethylation_expression.png")

Genes_hypomethylated_analysis %>% 
  dplyr::filter(., logFC > 0.75 & P.Value < 0.1) %>%
  .$ID %>%
  write.csv(., "../Results/Tables report/Gene lists/Genes_hypometh_surressed.csv", row.names = FALSE)

Genes_hypomethylated_analysis %>% 
  dplyr::filter(., logFC < -0.75 & P.Value < 0.1) %>%
  .$ID %>%
  write.csv(., "../Results/Tables report/Gene lists/Genes_hypometh_suppressed.csv", row.names = FALSE)
```

```{r}
Gene_methyl_enhancer_expression_data_target_gene <- dplyr::filter(Gene_methyl_enhancer_expression_data, ID %in% c(Genes_hypermethylated_analysis$ID, Genes_hypomethylated_analysis$ID))

nb_enhancer_treshold <- 1

Number_Genes_Hypermethylated_no_changes <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & (P.Value > 0.05 | abs(logFC) < 0.75)) %>%
  .$ID %>%
  length()
  
Number_Genes_Hypermethylated_enh_suppressed <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & P.Value < 0.05 & logFC < -0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypermethylated_enh_surrexpressed <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hyper > nb_enhancer_treshold & P.Value < 0.05 & logFC > 0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypomethylated_no_changes <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & (P.Value > 0.05 | abs(logFC)) < 0.75) %>%
  .$ID %>%
  length()
  
Number_Genes_Hypomethylated_enh_suppressed <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & P.Value < 0.05 & logFC < -0.75) %>%
  .$ID %>%
  length()

Number_Genes_Hypomethylated_enh_surrexpressed <- Gene_methyl_enhancer_expression_data_target_gene %>% 
  dplyr::filter(., nb_enhancer_hypo > nb_enhancer_treshold & P.Value < 0.05 & logFC > 0.75) %>%
  .$ID %>%
  length()

Numbers <- c(Number_Genes_Hypermethylated_enh_suppressed, 
             Number_Genes_Hypermethylated_no_changes, 
             Number_Genes_Hypermethylated_enh_surrexpressed, 
             Number_Genes_Hypomethylated_enh_suppressed, 
             Number_Genes_Hypomethylated_no_changes, 
             Number_Genes_Hypomethylated_enh_surrexpressed)


Barplot_data <- data.frame(Gene_expression=rep(c("down_regulated", "no_changes", "up_regulated"), 2),
                           Methylation=rep(c("Hypermethylated", "Hypomethylated"), each = 3),
                           Number_of_genes = Numbers)

ggplot(data=Barplot_data, aes(x=Methylation, y=Number_of_genes, fill=Gene_expression)) +
  geom_bar(stat="identity")
```


```{r}
methyl_enhancer_ge_corr <- data.frame(logFC = Gene_methyl_enhancer_expression_data$logFC,
                                      nb_enhancer_hypermethyl = Gene_methyl_enhancer_expression_data$nb_enhancer_hyper,
                                      gene = Gene_methyl_enhancer_expression_data$ID)

plot(x = methyl_enhancer_ge_corr$logFC_GE, y = methyl_enhancer_ge_corr$nb_enhancer_hypermethyl)
```


### logFC GE vs FC Met

```{r}
Genes_analysis <- Gene_methyl_expression_data_both_annotations %>%
  dplyr::filter(., Methylation_p.value < 0.05 & P.Value < 0.05)

methyl_ge_corr <- data.frame(logFC_M = Genes_analysis$delta_methylation,
                                    FC_GE = Genes_analysis$logFC,
                                    gene = Genes_analysis$ID)

methyl_ge_corr$logFC_GE <- ifelse(methyl_ge_corr$FC_GE > 0, log(methyl_ge_corr$FC_GE), -log(-methyl_ge_corr$FC_GE))

png("test2.png")
plot(x = methyl_ge_corr$logFC_M, y = methyl_ge_corr$logFC_GE)
dev.off()
plot(x = methyl_ge_corr$logFC_M, y = methyl_ge_corr$logFC_GE)
```

### Gene ontology analyse

```{r}
RNA_up_GO <- Gene_annotations %>% dplyr::filter(P.Value_expression < 0.05 & logFC_expression < -0.75) %>%
  .$Gene_name %>%
  unique() %>%
  enrichGO(keyType = "SYMBOL",
         OrgDb = "org.Hs.eg.db",
         ont = "BP",
         pAdjustMethod = "none",
         universe = dplyr::filter(Gene_annotations, P.Value_expression != 1)$Gene_name)

dotplot(RNA_up_GO, showCategory = 30)
```

```{r}
RNA_DM_prom_GO <- enrichGO(Genes_DM_promoter,
         keyType = "SYMBOL",
         OrgDb = "org.Hs.eg.db",
         ont = "BP",
         pAdjustMethod = "none",
         universe = dplyr::filter(Gene_annotations, P.Value_expression != 1 & P.Value_promoter_methylation != 1)$Gene_name)

dotplot(RNA_DM_prom_GO, showCategory = 30, title = "RNA_DM_prom_GO")
```


# After treatment

```{r}
A <- sapply(colnames(RNAseq_Wang_Feng), function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Non_Responder")
  x %in% tmp$Baseline_RNAseq_data
})

B <- sapply(colnames(RNAseq_Wang_Feng), function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Responder")
  x %in% tmp$Relapse_RNAseq_data
})

tmp_after <- data.frame("Non_responder" = A, "Responder" = B)
tmp_after$Phenotype <- ifelse(A, "Non_responder", ifelse(B, "Responder", "No_data"))

ann_color_RNA <- list(
    Phenotype = c(Responder = "blue", Non_responder = "red", No_data = "grey"))

png("../Results/Heatmap_RNAseq_relapse.png")
Make_heatmap(RNAseq_Wang_Feng, tmp_after, "pearson", "Non_responder & Responder", ann_color_RNA)
dev.off()

heatmap_Relapse <- Make_heatmap(RNAseq_Wang_Feng, tmp_after, "pearson", "Non_responder & Responder", ann_color_RNA)

colnames(RNAseq_Wang_Feng)[heatmap_Baseline$tree_col[["order"]]][c(1:15)]

batch = c(rep("A", 15), rep("B", 51-15))

combat_edata1 = ComBat(dat=RNAseq_Wang_Feng, batch=batch, mod=NULL, par.prior=TRUE, prior.plots=FALSE)

Make_heatmap(combat_edata1, tmp_after, "pearson", "Non_responder & Responder", ann_color_RNA, cuttree = 2)

RNAseq_Wang_Feng_After_analysis <- Differential_analysis(tmp_after$Phenotype, RNAseq_Wang_Feng)
```

# TF analysis

```{r}
RNAseq_Baselin_rownames <- RNAseq_Wang_Feng_Baseline$E %>% as.data.frame()


RNAseq_Baselin_rownames$Gene <- rownames(RNAseq_Baselin_rownames) %>% str_split(., "[|]") %>% lapply(., function(x){
  x[1]
}) %>%
  unlist(.)

RNAseq_Baselin_rownames <- RNAseq_Baselin_rownames %>%
  split(., .$Gene) %>%
  lapply(., function(x){
    l <- length(x[1,]) - 1
    cnames <- colnames(x)[c(1:l)]
    df <- x[,c(1:l)] %>%
      as.matrix(.) %>%
      colSums2(.) %>% 
      data.frame(.) %>%
      t(.) %>%
      data.frame(.)
    colnames(df) <- cnames
    df
  }) %>%
  rbindlist(.) %>% 
  data.frame(.)

gene_rownames <- rownames(RNAseq_Wang_Feng_Baseline) %>% str_split(., "[|]") %>% lapply(., function(x){
  x[1]
}) %>%
  unlist(.) %>% 
  unique(.)
rownames(RNAseq_Baselin_rownames) <- gene_rownames

data(dorothea_hs, package = "dorothea") 
regulons = dorothea_hs %>%
  filter(confidence %in% c("A", "B"))
tf_activities <- run_viper(RNAseq_Baselin_rownames, regulons, 
                           options =  list(method = "scale", minsize = 4, eset.filter = FALSE, cores = 1, verbose = FALSE))
tf_activities=t(tf_activities) %>% as.data.frame()
rownames(tf_activities) <- substr(rownames(tf_activities), 2, 11)
palette_length = 100
my_color = colorRampPalette(c("Darkblue", "white","red"))(palette_length)

Clustering_Baseline_phenotype <- read.csv("~/GitHub/Epigenomic_integration/DATA/Wang_Feng_DATA/Clustering_Baseline_phenotype.csv")

rownames(Clustering_Baseline_phenotype) <- Clustering_Baseline_phenotype$UPN

tmp$Cluster <- rownames(tmp) %>% 
  substr(., 1, 7) %>%
  Clustering_Baseline_phenotype[.,] %>%
  .$Cluster
  
tmp$Cluster[which(is.na(tmp$Cluster))] <- "No_data"

colnames(tmp) <- c("Non_Responder", "Responder", "Phenotype", "Cluster")

annotation_heatmap <- data.frame("Phenotype" = tmp$Phenotype)
rownames(annotation_heatmap) <- rownames(tmp)

ann_color_RNA <- list(
    Phenotype = c(Responder = "blue", Non_responder = "red")
    )

corr_TF <- rcorr(as.matrix(t(tf_activities)), type = "pearson")$r
png("../Results/tf_activities.png", height = 1080, width = 1920)
pheatmap(tf_activities, col = viridis::inferno(50), column_title = "Transcription Factor activities", annotation_row = annotation_heatmap, annotation_colors = ann_color_RNA)
dev.off()

png("../Results/Corr_patient_tf.png",  width = 360*(4/3), height = 360)
pheatmap(corr_TF, annotation_row = annotation_heatmap, annotation_col = annotation_heatmap, annotation_colors = ann_color_RNA, show_rownames = FALSE, show_colnames = FALSE, annotation_names_row = FALSE)
dev.off()
pheatmap(tf_activities, col = viridis::inferno(50), column_title = "Transcription Factor activities", annotation_row = annotation_heatmap, annotation_colors = ann_color_RNA)
pheatmap(corr_TF, annotation_row = annotation_heatmap, annotation_colors = ann_color_RNA)

```

```{r}
DATA_for_PCA <- RNAseq_Baselin_rownames
DATA_for_PCA <- DATA_for_PCA %>%
  t() %>%
  as.matrix()

res.pca <- PCA(DATA_for_PCA, graph = TRUE)

fviz_eig(res.pca, addlabels = TRUE)

fviz_pca_ind(res.pca,
             geom.ind = "point", # Montre les points seulement (mais pas le "text")
             col.ind = tmp$Phenotype, # colorer by groups
             palette = c("#00AFBB", "#E7B800", "#FC4E07"),
             addEllipses = TRUE, # Ellipses de concentration
             legend.title = "Groups"
             )

```


```{r}
A <- sapply(rownames(tf_activities), function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Non_Responder")
  x %in% tmp$Baseline_RNAseq_data
})

B <- sapply(rownames(tf_activities), function(x){
  tmp <- dplyr::filter(Phenotypes[["Koichi_cohort"]], Baseline_phenotype == "Responder")
  x %in% tmp$Baseline_RNAseq_data
})

test <- tf_activities[,c(1:2)]

tf_Ttest_activities <- tf_activities %>%
  sapply(., function(x){
    test_tmp <- t.test(x[A], x[B])
    c("p.value_activities" = test_tmp$p.value, 
               "Non_reponder_mean" = test_tmp$estimate[1], 
               "Responder" = test_tmp$estimate[2])
  }) %>%
  t()
  

write.csv(tf_Ttest_activities, file = "../Results/tf_Ttest_activities.csv")


RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$ID <- rownames(RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]) %>% str_split(., "[|]") %>% lapply(., function(x){
  x[1]
}) %>%
  unlist(.)

TF_activity_expression <- merge(tf_Ttest_activities, RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]], by.x = 0, by.y = "ID", all.x = TRUE)

TF_activity_expression$logFC <- ifelse(TF_activity_expression$logFC > 1, log(TF_activity_expression$logFC), ifelse(TF_activity_expression$logFC < -1 , -log(-TF_activity_expression$logFC), TF_activity_expression$logFC))

TF_activity_expression %>% .[order(.$p.value_activities),] %>% write.csv("../Results/TF_activities_expression.csv", row.names = FALSE)
```



```{r}
Non_responder_tf_activities <- tf_activities %>%
  .[rownames(tmp)[tmp$Phenotype == "Non_responder"],] %>%
  as.matrix(.) %>%
  colMeans2(.) %>%
  as.data.frame(.)

Responder_tf_activities <- tf_activities %>%
  .[rownames(tmp)[tmp$Phenotype == "Responder"],] %>%
  as.matrix(.) %>%
  colMeans2(.) %>%
  as.data.frame(.)

Tf_activities_phenotype <- data.frame("Non_Responder" = Non_responder_tf_activities, 
                                      "Responder" = Responder_tf_activities)
rownames(Tf_activities_phenotype) <- colnames(tf_activities)
colnames(Tf_activities_phenotype) <- c("Non_responder", "Responder")

Tf_activities_phenotype$delta <- Tf_activities_phenotype$Non_responder - Tf_activities_phenotype$Responder


write.csv(regulons, file = "../DATA/regulons.csv", row.names = FALSE)
write.csv(Tf_activities_phenotype, file = "../Results/Tf_activitiess_NR_R.csv")
```

# MS VIPER

```{r}
ref <- which(tmp$Phenotype == "Responder")

ref_name <- "Responder"
treat_name <- "Non_Responder"

msviper_results <- run_msviper(as.matrix(RNAseq_Baselin_rownames), regulons, use_aracne = T, ref, ref_name, treat_name, minsize = 4, ges.filter = T)

msviper_results$regulons$mor_cytoscape <- abs(msviper_results$regulons$mor)

msviper_results$regulons %>% write.csv("../Results/regulons_msviper.csv", row.names = FALSE)

original_gene_list <- RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$logFC

names(original_gene_list) <- RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$ID

gene_list<-na.omit(original_gene_list)

gene_list = gene_list[RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$P.Value<=0.05]

gene_list = sort(gene_list, decreasing = TRUE)

expre = data.frame("TF"=names(gene_list),"LogFC"=paste(gene_list)  )

msviper_results$mrs_table = left_join(msviper_results$mrs_table,expre,by="TF")
msviper_results$mrs_table= replace( msviper_results$mrs_table, is.na( msviper_results$mrs_table), '0') 
msviper_results$mrs_table$LogFC=  as.numeric(msviper_results$mrs_table$LogFC) 

msviper_results$mrs_table %>% write.csv("../Results/stats_on_TFs.csv")

all_nodes <- unique(c(msviper_results$regulons$tf, msviper_results$regulons$target))
tnodes <- tibble(TF = all_nodes)
all_nodes_metadata <- right_join(msviper_results$mrs_table, tnodes, by = "TF")
all_nodes_metadata$LogFC= RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$logFC[match(all_nodes_metadata$TF,RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]]$ID)]

all_nodes_metadata$LogFC <- ifelse(all_nodes_metadata$LogFC > 1, log(all_nodes_metadata$LogFC), ifelse(all_nodes_metadata$LogFC < -1 , -log(-all_nodes_metadata$LogFC), 0))

TF <- na.exclude(all_nodes_metadata$TF[all_nodes_metadata$pval<0.1]) %>% paste()

data_stat <- msviper_results$regulons[msviper_results$regulons$tf%in%TF,]

res_TF <- RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]] %>%
                  dplyr::filter(., ID %in% unique(data_stat$target))
res_TF <- res_TF[res_TF$P.Value<0.05,]

original_gene_list <- res_TF$logFC

# name the vector
names(original_gene_list) <- (rownames(res_TF))

# omit any NA values 
gene_list<-na.omit(original_gene_list)


gene_list <- sort(gene_list, decreasing = TRUE)

```


## Data for cytoscape

```{r}
Gene_annotations <- msviper_results$mrs_table
all_column_names <- c("Gene_name", "Size", "nes", "TF_Pval", "TF_Pval_fdr","TF_logFC")
colnames(Gene_annotations) <- all_column_names


Gene_annotations <- merge(Gene_annotations, Tf_activities_phenotype, by.x = "Gene_name", by.y = 0, all = TRUE)
all_column_names <- c(all_column_names, c("NR_TF_activity", "R_TF_activity", "delta_TF_activity"))
colnames(Gene_annotations) <- all_column_names


Gene_annotations <- merge(Gene_annotations, RNAseq_Wang_Feng_Baseline_analysis[["Non_responder-Responder"]], by.x = "Gene_name", by.y = "ID", all = TRUE) %>%
  dplyr::select(c(1:10,13))
all_column_names <- c(all_column_names, c("logFC_expression", "P.Value_expression"))
colnames(Gene_annotations) <- all_column_names

Gene_annotations <- merge(Gene_annotations, Results[["Basline_methylation_promoter"]]$Genes_methylation_analysis, by = "Gene_name", all = TRUE) 
all_column_names <- c(all_column_names, c("P.Value_promoter_methylation", "NR_Promoter_Methylation", "R_Promoter_Methylation", "delta_promoter_methylation"))
colnames(Gene_annotations) <- all_column_names

Gene_annotations <-  merge(Gene_annotations, Gene_enhancer_methylation_analysis, by.x = "Gene_name", by.y = 0, all = TRUE)

for (i in c(2,3, 6:10,13:19)){
  Gene_annotations[,i] <- ifelse(is.na(Gene_annotations[,i]), 0.0, Gene_annotations[,i])
}

for (i in c(4, 5,11,12)){
  Gene_annotations[,i] <- ifelse(is.na(Gene_annotations[,i]), 1.0, Gene_annotations[,i])
  }

Gene_annotations$nb_enhancer_changed <- Gene_annotations$nb_enhancer_hyper + Gene_annotations$nb_enhancer_hypo

write.csv(Gene_annotations, "../Results/Genes_values.csv", row.names = FALSE)
```

# Venn diagramms

```{r}
Genes_up_promoter <- Gene_annotations %>% 
  dplyr::filter(., P.Value_promoter_methylation != 1 & (logFC_expression > 0.75 & P.Value_expression < 0.05)) %>%
  .$Gene_name %>%
  unique()

Genes_down_promoter <- Gene_annotations %>% 
  dplyr::filter(., P.Value_promoter_methylation != 1 & (logFC_expression < -0.75 & P.Value_expression < 0.05)) %>%
  .$Gene_name %>% 
  unique()

Genes_DM_promoter <- Gene_annotations %>% 
  dplyr::filter(., P.Value_expression != 1 & P.Value_promoter_methylation < 0.05 & abs(delta_promoter_methylation) > 0.1) %>%
  .$Gene_name %>% 
  unique()

Genes_noDE_no_DM_promoter <- Gene_annotations %>% 
  dplyr::filter(., P.Value_promoter_methylation != 1 & P.Value_expression != 1 & (abs(logFC_expression) < 0.75 | P.Value_expression > 0.05) & (P.Value_promoter_methylation > 0.1 | abs(delta_promoter_methylation) < 0.1)) %>%
  .$Gene_name %>% 
  unique()

Venn_promoter_diagramm <- list(
  "DM" = Genes_DM_promoter,
  "Down" = Genes_down_promoter,
  "UP" = Genes_up_promoter
)
library(ggvenn)
ggvenn(
  Venn_promoter_diagramm, 
  fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF"))
ggsave("~/Venn_expression_Dmethylation_promoter.png")

table_fisher_test_promoter <- data.frame("Methylated_promoter" = 
                                           c(length(intersect(c(Genes_up_promoter, Genes_down_promoter), Genes_DM_promoter)),
                                             length(setdiff(Genes_DM_promoter, c(Genes_up_promoter, Genes_down_promoter)))), 
                                         "unmethylated_promoter" = 
                                           c(length(setdiff(c(Genes_up_promoter, Genes_down_promoter), Genes_DM_promoter)), 
                                             length(Genes_noDE_no_DM_promoter)))

rownames(table_fisher_test_promoter) <- c("DE", "no DE")

fisher.test(table_fisher_test_promoter)
```


```{r}
Genes_up_enhancer <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0 & logFC_expression > 0.75 & P.Value_expression < 0.05) %>%
  .$Gene_name %>%
  unique()

Genes_down_enhancer <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0 & logFC_expression < -0.75 & P.Value_expression < 0.05) %>%
  .$Gene_name %>% 
  unique()

Genes_DM_enhancer <- Gene_annotations %>% 
  dplyr::filter(.,  P.Value_expression != 1 & (nb_enhancer_hyper > 0 | nb_enhancer_hypo > 0)) %>%
  .$Gene_name %>%
  unique()

Genes_noDE_no_DM_enhancer <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0 & P.Value_expression != 1 & nb_enhancer_changed == 0 & (abs(logFC_expression) < 0.75 | P.Value_expression > 0.05)) %>%
  .$Gene_name %>% 
  unique()

Venn_enhancer_diagramm <- list(
  "DM_enhancer" = Genes_DM_enhancer,
  "Down" = Genes_down,
  "UP" = Genes_up
)

ggvenn(
  Venn_enhancer_diagramm, 
  fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF"))
ggsave("~/Venn_expression_Dmethylation_enhancer.png")

table_fisher_test_enhancer <- data.frame("Methylated_enhancer" = 
                                           c(length(intersect(c(Genes_up_enhancer, Genes_down_enhancer), Genes_DM_enhancer)),
                                             length(setdiff(Genes_DM_enhancer, c(Genes_up_enhancer, Genes_down_enhancer)))), 
                                         "unmethylated_enhancer" = 
                                           c(length(setdiff(c(Genes_up_enhancer, Genes_down_enhancer), Genes_DM_enhancer)), 
                                             length(Genes_noDE_no_DM_enhancer)))

rownames(table_fisher_test_enhancer) <- c("DE", "no DE")

fisher.test(table_fisher_test_enhancer)
```


```{r}
Genes_up_enhancer_promoter <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0 & P.Value_promoter_methylation != 1 & logFC_expression > 0.75 & P.Value_expression < 0.05) %>%
  .$Gene_name %>%
  unique()

Genes_down_enhancer_promoter <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0& P.Value_promoter_methylation != 1 & logFC_expression < -0.75 & P.Value_expression < 0.05) %>%
  .$Gene_name %>% 
  unique()

Genes_DM_enhancer_promoter <- Gene_annotations %>% 
  dplyr::filter(.,  P.Value_expression != 1 & (nb_enhancer_hyper > 0 | nb_enhancer_hypo > 0) & P.Value_promoter_methylation < 0.1 & abs(delta_promoter_methylation) > 0.1) %>%
  .$Gene_name %>%
  unique()

Genes_noDE_no_DM_enhancer_or_promoter <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0 & P.Value_expression != 1 & (nb_enhancer_changed == 0 | P.Value_promoter_methylation > 0.1 | abs(delta_promoter_methylation) < 0.1) & (abs(logFC_expression) < 0.75 | P.Value_expression > 0.05)) %>%
  .$Gene_name %>% 
  unique()

Venn_enhancer_promoter_diagramm <- list(
  "DM_enhancer_promoter" = Genes_DM_enhancer_promoter,
  "Down" = Genes_down_enhancer_promoter,
  "UP" = Genes_up_enhancer_promoter
)

ggvenn(
  Venn_enhancer_promoter_diagramm, 
  fill_color = c("#EFC000FF", "#868686FF", "#CD534CFF"))
ggsave("~/Venn_expression_Dmethylation_enhancer_promoter.png")

table_fisher_test_enhancer_promoter <- data.frame("Methylated_enhancer_promoter" = 
                                           c(length(intersect(c(Genes_up_enhancer_promoter, Genes_down_enhancer_promoter), Genes_DM_enhancer_promoter)),
                                             length(setdiff(Genes_DM_enhancer_promoter, c(Genes_up_enhancer_promoter, Genes_down_enhancer_promoter)))), 
                                         "unmethylated_enhancer" = 
                                           c(length(setdiff(c(Genes_up_enhancer_promoter, Genes_down_enhancer_promoter), Genes_DM_enhancer_promoter)), 
                                             length(Genes_noDE_no_DM_enhancer_or_promoter)))

rownames(table_fisher_test_enhancer_promoter) <- c("DE", "no DE")

fisher.test(table_fisher_test_enhancer_promoter)
```




```{r}
Genes_DM_enhancer_Down <- intersect(c(Genes_down_enhancer), Genes_DM_enhancer)

Genes_enhancer_unviverse <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0 & P.Value_expression != 1) %>%
  .$Gene_name %>% 
  unique()

Results[["Gene_ontology_Enhancer_methylation_and_Down"]] <- enrichGO(Genes_DM_enhancer_Down, 
         keyType = "SYMBOL",
         OrgDb = "org.Hs.eg.db",
         ont = "BP",
         pAdjustMethod = "none",
         universe = Genes_enhancer_unviverse)

ora_analysis_all_go_simplified <- clusterProfiler::simplify(Results[["Gene_ontology_Enhancer_methylation_and_Down"]]) 


svg(filename = "../Results/Gene_ontology_enhancers_DownExp_map.svg", width = 360*(16/9), height = 360)
emapplot(Results[["Gene_ontology_Enhancer_methylation_and_Down"]], color = "qvalue", size = "Count")
dev.off()
emapplot(Results[["Gene_ontology_Enhancer_methylation_and_Down"]], color = "qvalue", size = "Count")

dotplot(Results[["Gene_ontology_Enhancer_methylation_and_Down"]], showCategory = 30)
```

```{r}
Genes_DM_enhancer_DE <- intersect(c(Genes_down_enhancer, Genes_up_enhancer), Genes_DM_enhancer)

Genes_enhancer_unviverse <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0 & P.Value_expression != 1) %>%
  .$Gene_name %>% 
  unique()

Results[["Gene_ontology_Enhancer_methylation_and_DE"]] <- enrichGO(Genes_DM_enhancer_DE, 
         keyType = "SYMBOL",
         OrgDb = "org.Hs.eg.db",
         ont = "BP",
         pAdjustMethod = "none",
         universe = Genes_enhancer_unviverse)

ora_analysis_all_go_simplified <- clusterProfiler::simplify(Results[["Gene_ontology_Enhancer_methylation_and_DE"]]) 


#png(filename = "../Results/Gene_ontology_enhancers_DownExp_map.png", width = 360*(16/9), height = 360)
#emapplot(Results[["Gene_ontology_Enhancer_methylation_and_GE"]], color = "qvalue", size = "Count")
#dev.off()
emapplot(Results[["Gene_ontology_Enhancer_methylation_and_DE"]], color = "qvalue", size = "Count")

dotplot(Results[["Gene_ontology_Enhancer_methylation_and_DE"]], showCategory = 30)
```

```{r}
Genes_DM_enhancer_Up <- intersect(c(Genes_up_enhancer), Genes_DM_enhancer)

Genes_enhancer_unviverse <- Gene_annotations %>% 
  dplyr::filter(., total_enhancers > 0 & P.Value_expression != 1) %>%
  .$Gene_name %>% 
  unique()

Results[["Gene_ontology_Enhancer_methylation_and_UP"]] <- enrichGO(Genes_DM_enhancer_Up, 
         keyType = "SYMBOL",
         OrgDb = "org.Hs.eg.db",
         ont = "BP",
         pAdjustMethod = "none",
         universe = Genes_enhancer_unviverse)

ora_analysis_all_go_simplified_UP <- clusterProfiler::simplify(Results[["Gene_ontology_Enhancer_methylation_and_UP"]]) 


#png(filename = "../Results/Gene_ontology_enhancers_DownExp_map.png", width = 360*(16/9), height = 360)
#emapplot(Results[["Gene_ontology_Enhancer_methylation_and_GE"]], color = "qvalue", size = "Count")
#dev.off()
emapplot(Results[["Gene_ontology_Enhancer_methylation_and_UP"]], color = "qvalue", size = "Count")

dotplot(Results[["Gene_ontology_Enhancer_methylation_and_UP"]], showCategory = 30)
```

```{r}
Genes_metabo_DM_enhancer <- c("COX6A1","COX7A2L","NDUFA4", "STM3","MGST1","MGST3")

RNA_selected_genes <- RNAseq_Baselin_rownames[which(rownames(RNAseq_Baselin_rownames) %in% Genes_metabo_DM_enhancer),]
colnames(RNA_selected_genes) <- substring(colnames(RNA_selected_genes), 2,11)

png("../Results/Clustering_Baseline_based_on_gene_metabo_DM_enhancer.png", width = 360*(16/9), height = 360)
Make_heatmap(RNA_selected_genes, Phenotype = tmp, "pearson", "Non_responder & Responder based on Genes metabo DM enhancer", ann_color_RNA)
dev.off()
```

