---
title: "R Notebook"
output: html_notebook
---

```{r, include=FALSE}
load("../DATA/CD34/Methylation/BMIQ_CD34.Rdata")
load("../DATA/IDHm/Methylation/BMIQ_IDHm.RData")
load("../DATA/BMIQ_all.RData")
load("../DATA/Methyl_DATA_Wang_Feng.RData")
load("../DATA/DATA_whiele.RData")
load("../DATA/Wang_Feng_RNAseq.RData")


gene_list <- c("CEBPA", "CPT1A", "CPT2", "SLC25A20", "AKT2", "PPARGC1A", "PTEN")
```

```{r}
Blueprint <- read.csv("../BLUEPRINT_fragments_good.tsv", sep = "\t") %>%
  dplyr::select(., "gene_names":"type", "gene_type") %>%
  separate_rows(., gene_names, sep = " ") %>%
  separate_rows(., gene_type, sep = " ") %>%
  unique(.)

anno <- read.csv("~/Illumina_Manifest/HumanMethylation450_15017482_v1-2.csv", skip = 7) %>% 
  dplyr::select(., "Name", "CHR", "MAPINFO", "UCSC_RefGene_Name", "UCSC_RefGene_Group", "Relation_to_UCSC_CpG_Island") %>%
  dplyr::filter(., CHR != "") %>%
  separate_rows(., UCSC_RefGene_Name, UCSC_RefGene_Group, sep = ";")

Blueprint_Granges <- GRanges(
    seqnames = Blueprint$chr,
    ranges = IRanges(start = Blueprint$start, end = Blueprint$end),
    Blueprint_gene_names = Blueprint$gene_names,
    type = Blueprint$type,
    gene_type = Blueprint$gene_type
  )
CpGs_Granges <- GRanges(
  seqnames = anno$CHR,
  ranges = IRanges(anno$MAPINFO, anno$MAPINFO +1),
  CpG = anno$Name,
  Illumina_Gene_name = anno$UCSC_RefGene_Name,
  position = anno$UCSC_RefGene_Group,
  Island = anno$Relation_to_UCSC_CpG_Island
)
overlaps_CpGs_Blueprint <- findOverlaps(Blueprint_Granges, CpGs_Granges)
match_hit_CpGs_Blueprint <- data.frame(mcols(Blueprint_Granges[queryHits(overlaps_CpGs_Blueprint),]),
                                       data.frame(mcols(CpGs_Granges[subjectHits(overlaps_CpGs_Blueprint),])))

match_hit_CpGs_Blueprint_promoter <- dplyr::filter(match_hit_CpGs_Blueprint, type == "P")

anno_promoter <- dplyr::filter(anno, UCSC_RefGene_Group == "TSS1500" | UCSC_RefGene_Group == "TSS200")

pchic <- prepare_pchic()
PCHiC_bed <- unique(rbind(pchic[, c(1:3, 5)], pchic[, c(6:8, 10)]))
PCHiC_GRange <- GRanges(
  seqnames = PCHiC_bed$chr,
  IRanges(start = PCHiC_bed$start, end = PCHiC_bed$end),
  Gene_Pchic = PCHiC_bed$Name,
  start_fragment = PCHiC_bed$start,
  end_fragment = PCHiC_bed$end
)
PCHiC_GRange$ID <- paste(PCHiC_bed$chr, PCHiC_bed$start, sep = "_")
colnames(pchic) <- c("chr_bait", "start_bait", "end_bait", "ID_bait", "Name_bait", "chr_oe", "start_oe", "end_oe", "ID_oe", "Name_oe")
pchic$IDbait <- paste(pchic$chr_bait, pchic$start_bait, sep = "_")
pchic$IDoe <- paste(pchic$chr_oe, pchic$start_oe, sep = "_")

overlaps_CpGs_Pchic <- findOverlaps(CpGs_Granges, PCHiC_GRange)
matchit_CpGs_Pchic <- data.frame(mcols(CpGs_Granges[queryHits(overlaps_CpGs_Pchic),]),
                                       data.frame(mcols(PCHiC_GRange[subjectHits(overlaps_CpGs_Pchic),])))

overlaps_Blueprint_Pchic <- findOverlaps(Blueprint_Granges, PCHiC_GRange)
matchit_Blueprint_Pchic <- data.frame(mcols(Blueprint_Granges[queryHits(overlaps_Blueprint_Pchic),]),
                                       data.frame(mcols(PCHiC_GRange[subjectHits(overlaps_Blueprint_Pchic),]))) %>%
  dplyr::filter(., type == "P")


```





```{r}
Phenotype_FLT3_IDH_CD34 <- read.csv("../../TCGA_Connections/Phenotype_met.csv") %>%
  dplyr::filter(., WT1 == "WT1WT", DNMT3A == "DNMT3AWT", TET2 == "TET2WT")

Phenotype_FLT3_IDH_CD34$Phenotype <- ifelse(Phenotype_FLT3_IDH_CD34$IDH == "IDHm", "IDHm", 
                                   ifelse(Phenotype_FLT3_IDH_CD34$FLT3 == "FLT3m", "FLT3m", "WT"))
  
Phenotype_FLT3_IDH_CD34 <- dplyr::select(Phenotype_FLT3_IDH_CD34, c(X, Phenotype)) %>%
  rbind(data.frame(X = colnames(BMIQ_CD34), Phenotype = rep("CD34", 6)))

Phenotype_FLT3_IDH_CD34$X <- str_replace(Phenotype_FLT3_IDH_CD34$X, "-", ".")
Phenotype_FLT3_IDH_CD34$X <- str_replace(Phenotype_FLT3_IDH_CD34$X, "-", ".")

DATA_met_CD34_IDH_FLT3 <- BMIQ_all %>%
  dplyr::select(., Phenotype_FLT3_IDH_CD34$X)


Phenotype_BMIQ_CD34_IDHm_WT <- read.csv("../../TCGA_Connections/Phenotype_met.csv") %>%
  dplyr::filter(., WT1 == "WT1WT", DNMT3A == "DNMT3AWT", TET2 == "TET2WT", FLT3 == "FLT3AWT")
Phenotype_BMIQ_CD34_IDHm_WT$Phenotype <- ifelse(Phenotype_BMIQ_CD34_IDHm_WT$IDH == "IDHm", "IDHm", "WT")

Phenotype_BMIQ_CD34_IDHm_WT <- dplyr::select(Phenotype_BMIQ_CD34_IDHm_WT, c(X, Phenotype)) %>%
  rbind(data.frame(X = colnames(BMIQ_CD34), Phenotype = rep("CD34", 6)))

Phenotype_BMIQ_CD34_IDHm_WT$X <- str_replace(Phenotype_BMIQ_CD34_IDHm_WT$X, "-", ".")
Phenotype_BMIQ_CD34_IDHm_WT$X <- str_replace(Phenotype_BMIQ_CD34_IDHm_WT$X, "-", ".")

BMIQ_CD34_IDHm_WT <- BMIQ_all %>%
  dplyr::select(., Phenotype_BMIQ_CD34_IDHm_WT$X)

```


#### Looking at global methylation on promoter of specific genes through CD34 IDHm and IDHWT
```{r}
for (i in gene_list){
  Focus_Gene_global_methylation(i, match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_BMIQ_CD34_IDHm_WT, BMIQ_CD34_IDHm_WT, c("IDHm", "WT"), c("#00FF00", "#FF0000", "#0000FF"))
}

Focus_Gene_global_methylation("", match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_BMIQ_CD34_IDHm_WT, BMIQ_CD34_IDHm_WT, c("IDHm", "WT"), c("#00FF00", "#FF0000", "#0000FF"))
```
#### Looking at global methylation on neighbor of promoter of specific genes through CD34 IDHm and IDHWT
```{r}
for (i in gene_list){
  Focus_gene_promoter_neighborhood_methylation(i, match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_BMIQ_CD34_IDHm_WT, BMIQ_CD34_IDHm_WT, c("IDHm", "WT"), c("#00FF00", "#FF0000", "#0000FF"), matchit_CpGs_Pchic, pchic, match_hit_CpGs_Blueprint)
}
```





```{r}
efit.contrast <- Look_at_differential_methylation(Phenotype_TCGA, DATA_met_CD34_IDH_FLT3)

print(colnames(efit.contrast[["coefficients"]]))

filtered_DMP_IDHmvsWT <- Filter_comparison_methylation("IDHm_vs_WT", DATA_met_CD34_IDH_FLT3, efit.contrast)

filtered_DMP_FLT3vsWT <- Filter_comparison_methylation("FLT3m_vs_WT", DATA_met_CD34_IDH_FLT3, efit.contrast)

filtered_DMP_WTvsCD34 <- Filter_comparison_methylation("WT_vs_CD34", DATA_met_CD34_IDH_FLT3, efit.contrast)

filtered_DMP_IDHmvsCD34 <- Filter_comparison_methylation("IDHm_vs_CD34", DATA_met_CD34_IDH_FLT3, efit.contrast)

```

```{r}
for (i in gene_list){
  Focus_Gene_global_methylation(i, match_hit_CpGs_Blueprint, anno_promoter, Phenotype_no_CD34, BMIQ_met, c("IDHm", "IDHWT"), c("#FF0000", "#0000FF"))
}

CpGs_DM <- rownames(filtered_DMP_IDHmvsIDHWT)
  
for (i in gene_list){
  Focus_Gene_differential_methylation(i, match_hit_CpGs_Blueprint, anno_promoter, Phenotype_no_CD34, BMIQ_met, CpGs_DM, c("IDHm", "IDHWT"), c("#FF0000", "#0000FF"))
}


```


#### DATA FENG WANG
```{r}
Phenotype_Wang_Feng_Control_Baseline <- Phenotype_Wang_Feng %>%
  dplyr::filter(., Phenotype == "Baseline" | Phenotype == "Control")
  
BMIQ_Control_Baseline <- dplyr::select(BMIQ_Wang_Feng, Phenotype_Wang_Feng_Control_Baseline$Sample_number)

ann_colors <- list(
    Phenotype = c(Baseline = "blue", Responder = "green", Non_Responder = "red", Control = "grey"))
annotation_for_heatmap <- data.frame(Phenotype = Phenotype_Wang_Feng_Control_Baseline$Phenotype)
rownames(annotation_for_heatmap) <- colnames(BMIQ_Control_Baseline)
corr_spearman_Control_Baseline <- rcorr(as.matrix(BMIQ_Control_Baseline), type = "spearman")$r
colnames(corr_spearman_Control_Baseline) <- colnames(BMIQ_Control_Baseline)
heatmap_corr_spearman_Control_Baseline <- pheatmap(corr_spearman_Control_Baseline, 
       color = colorRampPalette(brewer.pal(n = 9, name = "YlOrRd"))(100),
       annotation_col = annotation_for_heatmap,
       annotation_colors = ann_colors,
       legend = TRUE,
       treeheight_row = 20,
       main = "corr_spearman_Control_Baseline", 
       fontsize = 10
       )
png(filename = paste0("../Results/Wang_Feng_Results/heatmap_corr_spearman_Control_Baseline.png"), width = 960, height = 540)
heatmap_corr_spearman_Control_Baseline
dev.off()






Phenotype_Wang_Feng_Responder_Non_Responder <- Phenotype_Wang_Feng %>%
  dplyr::filter(., Phenotype != "Baseline" & Phenotype != "Control")

BMIQ_Responder_Non_Responder <- dplyr::select(BMIQ_Wang_Feng, Phenotype_Wang_Feng_Responder_Non_Responder$Sample_number)

annotation_for_heatmap <- data.frame(Phenotype = Phenotype_Wang_Feng_Responder_Non_Responder$Phenotype)
rownames(annotation_for_heatmap) <- colnames(BMIQ_Responder_Non_Responder)
corr_pearson_Responder_Non_Responder <- rcorr(as.matrix(BMIQ_Responder_Non_Responder), type = "pearson")$r
colnames(corr_pearson_Responder_Non_Responder) <- colnames(BMIQ_Responder_Non_Responder)
heatmap_corr_pearson_Responder_Non_Responder <- pheatmap(corr_pearson_Responder_Non_Responder, 
       color = colorRampPalette(brewer.pal(n = 9, name = "YlOrRd"))(100),
       annotation_col = annotation_for_heatmap,
       annotation_colors = ann_colors,
       legend = TRUE,
       treeheight_row = 20,
       main = "corr_pearson_Responder_Non_Responder", 
       fontsize = 10
       )
png(filename = paste0("../Results/Wang_Feng_Results/heatmap_corr_pearson_Responder_Non_Responder.png"), width = 960, height = 540)
heatmap_corr_pearson_Responder_Non_Responder
dev.off()

png(filename = paste0("../Results/Wang_Feng_Results/heatmap_spearman_BMIQ_Wang_Feng.png"), width = 960, height = 540)
heatmap_spearman
dev.off()
png(filename = paste0("../Results/Wang_Feng_Results/heatmap_pearson_BMIQ_Wang_Feng.png"), width = 960, height = 540)
heatmap_pearson
dev.off()



```

```{r}
for (i in gene_list){
  Focus_Gene_global_methylation(i, match_hit_CpGs_Blueprint, anno_promoter, Phenotype_Wang_Feng, BMIQ_Wang_Feng, c("Baseline", "Control"), c("#0000FF", "#555555", "#FF0000", "#00FF00"), Change_directory = "Wang_Feng_Results/")
}
```

```{r}
for (i in gene_list){
  Focus_gene_promoter_neighborhood_methylation(i, match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_Wang_Feng, BMIQ_Wang_Feng, c("Baseline", "Control"), c("#0000FF", "#555555", "#FF0000", "#00FF00"), matchit_CpGs_Pchic, pchic, match_hit_CpGs_Blueprint, Change_directory = "Wang_Feng_Results/")
}
```





```{r}
for (i in gene_list){
  Focus_Gene_global_methylation(i, match_hit_CpGs_Blueprint, anno_promoter, Phenotype_Whiele, BMIQ_Whiele, c("IDH2m", "WT"), c("#555555", "#FF0000", "#00FF00", "#0000FF"), Change_directory = "Whiele_Results/")
}

Focus_Gene_global_methylation("CEBPA", match_hit_CpGs_Blueprint, anno_promoter, Phenotype_Whiele, BMIQ_Whiele, c("IDH2m", "WT"), c("#555555", "#FF0000", "#0000FF", "#00FF00"), Change_directory = "Whiele_Results/")


```

```{r}
for (i in gene_list){
  Focus_gene_promoter_neighborhood_methylation(i, match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_Whiele, BMIQ_Whiele, c("IDH2m","WT"), c("#555555", "#FF0000", "#00FF00", "#0000FF"), matchit_CpGs_Pchic, pchic, match_hit_CpGs_Blueprint, Change_directory = "Whiele_Results/")
}
```


```{r}
Phenotype_noCD34 <- Phenotype_BMIQ_CD34_IDHm_WT %>%
  dplyr::filter(., Phenotype !="CD34")

BMIQ_noCD34 <- BMIQ_all %>%
  dplyr::select(., Phenotype_noCD34$X)

for (i in gene_list){
  Focus_Gene_global_methylation(i, match_hit_CpGs_Blueprint, anno_promoter, Phenotype_noCD34, BMIQ_noCD34, c("IDHm", "WT"), c("#FF0000", "#0000FF"), Change_directory = "NoCD34/")
}
```
```{r}
for (i in gene_list){
  Focus_gene_promoter_neighborhood_methylation(i, match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_noCD34, BMIQ_noCD34, c("IDHm","WT"), c("#FF0000", "#0000FF"), matchit_CpGs_Pchic, pchic, match_hit_CpGs_Blueprint, Change_directory = "NoCD34/")
}
```

```{r}
Focus_Gene_global_methylation("CEBPA", match_hit_CpGs_Blueprint, anno_promoter, Phenotype_noCD34, BMIQ_noCD34, c("IDHm", "WT"), c("#FF0000", "#0000FF"), Change_directory = "NoCD34/")
```


## RNAseq_Wang_Feng
```{r}
ann_colors <- list(
    Phenotype = c(Baseline = "blue", Relapse = "red"))
annotation_for_heatmap <- data.frame(Phenotype = Phenotype_Wang_Feng_RNAseq$Phenotype)
rownames(annotation_for_heatmap) <- colnames(RNAseq_Wang_Feng)
corr_spearman_Baseline_Relapse <- rcorr(as.matrix(RNAseq_Wang_Feng), type = "spearman")$r
colnames(corr_spearman_Baseline_Relapse) <- colnames(RNAseq_Wang_Feng)
heatmap_corr_spearman_Baseline_Relapse <- pheatmap(corr_spearman_Baseline_Relapse, 
       color = colorRampPalette(brewer.pal(n = 9, name = "YlOrRd"))(100),
       annotation_col = annotation_for_heatmap,
       annotation_colors = ann_colors,
       legend = TRUE,
       treeheight_row = 20,
       main = "corr_spearman_Baseline_Relapse", 
       fontsize = 10
       )

```

```{r}
RNAseq_Wang_Feng_analysis <- Differential_analysis(Phenotype_Wang_Feng_RNAseq$Phenotype, RNAseq_Wang_Feng, "gene expression")

Baseline_Relapse_gene_expression_analysis <- RNAseq_Wang_Feng_analysis[["Baseline-Relapse"]] %>%
  dplyr::filter(., )


up_entrez_in_Baseline <- Baseline_Relapse_gene_expression_analysis %>%
  dplyr::filter(., logFC > 0 & P.Value < 0.05) %>%
  separate_rows(., ID, sep = "[|]") %>%
  dplyr::filter(., grepl("[A-Za-z]", ID))
down_entrez_in_Baseline <- Baseline_Relapse_gene_expression_analysis %>%
  dplyr::filter(., logFC < 0 & P.Value < 0.05) %>%
  separate_rows(., ID, sep = "[|]") %>%
  dplyr::filter(., grepl("[A-Za-z]", ID))




message("========================================================")
message("[-----------------Comparison starting-------------------]")
ego_up <- enrichGO(
  gene = unique(up_entrez$ID),
  keyType = "SYMBOL",
  OrgDb = "org.Hs.eg.db",
  ont = "ALL",
  pAdjustMethod = "none"
)
ego_down <- enrichGO(
  gene = unique(down_entrez$ID),
  keyType = "SYMBOL",
  OrgDb = "org.Hs.eg.db",
  ont = "ALL",
  pAdjustMethod = "none"
)

dotplot(ego_up, showCategory = 30)
dotplot(ego_down, showCategory = 30)

write.csv(ego_up@result, "../Results/GO_analysis_Genes_up_Baseline_vs_Relapse.csv")
write.csv(ego_down@result, "../Results/GO_analysis_Genes_down_Baseline_vs_Relapse.csv")

```
```{r}
ensembl = useMart("ensembl", dataset = "hsapiens_gene_ensembl")

up_entrez_in_Baseline_annotated <- add_genes_annotations(up_entrez_in_Baseline, "hgnc_symbol") %>%
  dplyr::filter(., !is.na(ensembl_gene_id))
down_entrez_in_Baseline_annotated <- add_genes_annotations(down_entrez_in_Baseline, "hgnc_symbol") %>%
  dplyr::filter(., !is.na(ensembl_gene_id))

write.csv(up_entrez_in_Baseline_annotated, "../Results/Genes_Up_in_Baseline_vs_Relapse.csv")
write.csv(down_entrez_in_Baseline_annotated, "../Results/Genes_Down_in_Baseline_vs_Relapse.csv")

```


```{r}
Genes_up <- up_entrez$ID %>%
  sapply(., function(gene) {
    ifelse(length(Gene_name_finding(gene)) >0, TRUE, FALSE)
  }) %>%
  up_entrez$ID[.]

Genes_down <- down_entrez$ID %>%
  sapply(., function(gene) {
    ifelse(length(Gene_name_finding(gene)) >0, TRUE, FALSE)
  }) %>%
  down_entrez$ID[.]

```



```{r}
signature <- read.csv("../DATA/BPmetCan.txt", sep = "\t")
rownames(signature) <- signature$CpGs
signature <- signature[,-1]

Baseline_sample <- Phenotype_Wang_Feng %>% 
  .[.$Phenotype == "Baseline", "Sample_number"]

res_Baseline <- epidish(as.matrix(BMIQ_Wang_Feng %>%
                                    dplyr::select(., Baseline_sample)), as.matrix(signature), method = "RPC")
Fres_Baseline <- as.data.frame(res_Baseline$estF)
Fres_Baseline <- cbind(Sample = rownames(Fres_Baseline), Fres_Baseline)
Fres_Baseline[, -1] <- round(Fres_Baseline[, -1], 3)

deconvolution_violin_Baseline <- data.frame(Cell_Type = rep(colnames(Fres_Baseline[,c(2:12)]), each = length(rownames(Fres_Baseline))),
                          Values = Fres_Baseline[,c(2:12)] %>% c(.) %>% unlist(.) %>% as.vector(.))
deconvolution_violin_Baseline$Values <- round(deconvolution_violin_Baseline$Values, 1)

Responder_sample <- Phenotype_Wang_Feng %>% 
  .[.$Phenotype == "Responder", "Sample_number"]

res_Responder <- epidish(as.matrix(BMIQ_Wang_Feng %>%
                                    dplyr::select(., Responder_sample)), as.matrix(signature), method = "RPC")
Fres_Responder <- as.data.frame(res_Responder$estF)
Fres_Responder <- cbind(Sample = rownames(Fres_Responder), Fres_Responder)
Fres_Responder[, -1] <- round(Fres_Responder[, -1], 3)

deconvolution_violin_Responder <- data.frame(Cell_Type = rep(colnames(Fres_Responder[,c(2:12)]), each = length(rownames(Fres_Responder))),
                          Values = Fres_Responder[,c(2:12)] %>% c(.) %>% unlist(.) %>% as.vector(.))
deconvolution_violin_Responder$Values <- round(deconvolution_violin_Responder$Values, 1)


Non_Responder_sample <- Phenotype_Wang_Feng %>% 
  .[.$Phenotype == "Non_Responder", "Sample_number"]

res_Non_Responder <- epidish(as.matrix(BMIQ_Wang_Feng %>%
                                    dplyr::select(., Non_Responder_sample)), as.matrix(signature), method = "RPC")
Fres_Non_Responder <- as.data.frame(res_Non_Responder$estF)
Fres_Non_Responder <- cbind(Sample = rownames(Fres_Non_Responder), Fres_Non_Responder)
Fres_Non_Responder[, -1] <- round(Fres_Non_Responder[, -1], 3)

deconvolution_violin_Non_Responder <- data.frame(Cell_Type = rep(colnames(Fres_Non_Responder[,c(2:12)]), each = length(rownames(Fres_Non_Responder))),
                          Values = Fres_Non_Responder[,c(2:12)] %>% c(.) %>% unlist(.) %>% as.vector(.))
deconvolution_violin_Non_Responder$Values <- round(deconvolution_violin_Non_Responder$Values, 1)




Control_sample <- Phenotype_Wang_Feng %>% 
  .[.$Phenotype == "Control", "Sample_number"]

res_Control <- epidish(as.matrix(BMIQ_Wang_Feng %>%
                                    dplyr::select(., Control_sample)), as.matrix(signature), method = "RPC")
Fres_Control <- as.data.frame(res_Control$estF)
Fres_Control <- cbind(Sample = rownames(Fres_Control), Fres_Control)
Fres_Control[, -1] <- round(Fres_Control[, -1], 3)

deconvolution_violin_Control <- data.frame(Cell_Type = rep(colnames(Fres_Control[,c(2:12)]), each = length(rownames(Fres_Control))),
                          Values = Fres_Control[,c(2:12)] %>% c(.) %>% unlist(.) %>% as.vector(.))
deconvolution_violin_Control$Values <- round(deconvolution_violin_Control$Values, 1)
```


```{r}
p <- ggplot(deconvolution_violin_Baseline, aes(x=Cell_Type, y=Values, fill=Cell_Type), las = 2) + 
  geom_violin() + 
  ggtitle("Baseline")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
ggsave(
    filename = "../Results/Wang_Feng_Results/Deconv/Baseline.png",
    plot = last_plot(),
    device = NULL,
    path = NULL,
    scale = 1,
    width = NA,
    height = NA,
    units = c("in", "cm", "mm"),
    dpi = 300,
    limitsize = TRUE
  )

p <- ggplot(deconvolution_violin_Responder, aes(x=Cell_Type, y=Values, fill=Cell_Type), las = 2) + 
  geom_violin() + 
  ggtitle("Responder")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
ggsave(
    filename = "../Results/Wang_Feng_Results/Deconv/Responder.png",
    plot = last_plot(),
    device = NULL,
    path = NULL,
    scale = 1,
    width = NA,
    height = NA,
    units = c("in", "cm", "mm"),
    dpi = 300,
    limitsize = TRUE
  )

p <- ggplot(deconvolution_violin_Non_Responder, aes(x=Cell_Type, y=Values, fill=Cell_Type), las = 2) + 
  geom_violin() + 
  ggtitle("Non_Responder")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
ggsave(
    filename = "../Results/Wang_Feng_Results/Deconv/Non_Responder.png",
    plot = last_plot(),
    device = NULL,
    path = NULL,
    scale = 1,
    width = NA,
    height = NA,
    units = c("in", "cm", "mm"),
    dpi = 300,
    limitsize = TRUE
  )


p <- ggplot(deconvolution_violin_Control, aes(x=Cell_Type, y=Values, fill=Cell_Type), las = 2) + 
  geom_violin() + 
  ggtitle("Control")+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
ggsave(
    filename = "../Results/Wang_Feng_Results/Deconv/Control.png",
    plot = last_plot(),
    device = NULL,
    path = NULL,
    scale = 1,
    width = NA,
    height = NA,
    units = c("in", "cm", "mm"),
    dpi = 300,
    limitsize = TRUE
  )
```

```{r}
BMIQ_Wang_Feng_high_variance <- Focus_high_variance_CpGs(BMIQ_Wang_Feng, percentage = 10)

Focus_global_methylation(match_hit_CpGs_Blueprint, anno, Phenotype_Wang_Feng, BMIQ_Wang_Feng_high_variance, c("Baseline", "Control"), fill_colour = c("#0000FF", "#777777", "#FF0000", "#00FF00"), Change_directory = "High_variance/")

Focus_global_methylation(match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_Wang_Feng, BMIQ_Wang_Feng_high_variance, c("Baseline", "Control"), fill_colour = c("#0000FF", "#777777", "#FF0000", "#00FF00"), Change_directory = "High_variance/Promoter/")
```

```{r}
BMIQ_all_high_variance <- Focus_high_variance_CpGs(BMIQ_CD34_IDHm_WT, percentage = 1)

Focus_global_methylation(match_hit_CpGs_Blueprint, anno, Phenotype_BMIQ_CD34_IDHm_WT, BMIQ_all_high_variance, c("IDHm", "WT"), fill_colour = c("#00FF00", "#FF0000", "#0000FF", "#00FF00"), Change_directory = "High_variance/")

Focus_global_methylation(match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_BMIQ_CD34_IDHm_WT, BMIQ_all_high_variance, c("IDHm", "WT"), fill_colour = c("#00FF00", "#FF0000", "#0000FF", "#00FF00"), Change_directory = "High_variance/Promoter/")

```

```{r}
Phenotype_Wang_Feng_RNAseq$Sample_name <- Phenotype_Wang_Feng_RNAseq$Sample %>%
  str_remove(., "_REL") %>%
  str_remove(., "_BL")

Phenotype_Wang_Feng_RNAseq
```

```{r}
ann_colors <- list(
    Phenotype = c(Baseline = "blue", Non_Responder = "red", Responder = "green", Control = "grey"))
annotation_for_heatmap <- data.frame(Phenotype = Phenotype_Wang_Feng$Phenotype)
rownames(annotation_for_heatmap) <- colnames(BMIQ_Wang_Feng_high_variance)
corr_spearman_Baseline_Non_Responder_Control  <- rcorr(as.matrix(BMIQ_Wang_Feng_high_variance), type = "spearman")$r
colnames(corr_spearman_Baseline_Non_Responder_Control ) <- colnames(BMIQ_Wang_Feng_high_variance)
heatmap_corr_spearman_Baseline_Non_Responder_Control <- pheatmap(corr_spearman_Baseline_Non_Responder_Control , 
       color = colorRampPalette(brewer.pal(n = 9, name = "YlOrRd"))(100),
       annotation_col = annotation_for_heatmap,
       annotation_colors = ann_colors,
       legend = TRUE,
       treeheight_row = 20,
       main = "corr_spearman_Baseline_Relapse", 
       fontsize = 10
       )



```

```{r}
Phenotype_Wang_Feng_Control_Baseline <- Phenotype_Wang_Feng %>%
  dplyr::filter(., Phenotype == "Baseline" | Phenotype == "Control")
  
BMIQ_Control_Baseline <- dplyr::select(BMIQ_Wang_Feng_high_variance, Phenotype_Wang_Feng_Control_Baseline$Sample_number)

Phenotype_Wang_Feng_Responder_Non_Responder <- Phenotype_Wang_Feng %>%
  dplyr::filter(., Phenotype != "Baseline" & Phenotype != "Control")

BMIQ_Responder_Non_Responder <- dplyr::select(BMIQ_Wang_Feng_high_variance, Phenotype_Wang_Feng_Responder_Non_Responder$Sample_number)


heatmap_Control_vs_Baseline_spearman <- Make_heatmap(BMIQ_Control_Baseline, Phenotype_Wang_Feng_Control_Baseline, method = "spearman", title = "Control vs Baseline", annotation_color = ann_colors)

heatmap_Control_vs_Baseline_pearson <- Make_heatmap(BMIQ_Control_Baseline, Phenotype_Wang_Feng_Control_Baseline, method = "pearson", title = "Control vs Baseline", annotation_color = ann_colors)

heatmap_Responder_vs_Non_Responder_spearman <- Make_heatmap(BMIQ_Responder_Non_Responder, Phenotype_Wang_Feng_Responder_Non_Responder, method = "spearman", title = "Responder vs Non Responder", annotation_color = ann_colors)

heatmap_Responder_vs_Non_Responder_pearson <- Make_heatmap(BMIQ_Responder_Non_Responder, Phenotype_Wang_Feng_Responder_Non_Responder, method = "pearson", title = "Responder vs Non Responder", annotation_color = ann_colors)
```


```{r}
Phenotype_cluster <- read.csv("../DATA/Wang_Feng_DATA/Clustering_Baseline_phenotype.csv")

Phenotype_cluster_DNAmet <- merge(Phenotype_cluster, Phenotype_Wang_Feng, by.x = "UPN", by.y = "Sample", all.y = TRUE)

BMIQ_wang_feng_analysis <- Differential_analysis(Phenotype_Wang_Feng$Phenotype, BMIQ_Wang_Feng)

Response_diff_hypermet_in_Non_Responder <- BMIQ_wang_feng_analysis[["Non_Responder-Responder"]] %>% 
  dplyr::filter(., P.Value < 0.01) %>%
  dplyr::filter(., logFC > 0.3)

Response_diff_hypermet_in_Responder <- BMIQ_wang_feng_analysis[["Non_Responder-Responder"]] %>% 
  dplyr::filter(., P.Value < 0.01) %>%
  dplyr::filter(., logFC < 0.3)


Genes_hypermetylated_in_Non_responder <- Look_at_gene_with_CpGs_in_promoter(Response_diff_hypermet_in_Non_Responder$ID, match_hit_CpGs_Blueprint_promoter)

Genes_hypermetylated_in_Non_responder_enhancer <- Look_at_genes_connected_to_CpGs(Response_diff_hypermet_in_Non_Responder$ID,
                                                                                  matchit_CpGs_Pchic, 
                                                                                  pchic,
                                                                                  matchit_Blueprint_Pchic,
                                                                                  match_hit_CpGs_Blueprint_promoter)

```
## Trouver les gènes differement methylés entre control et Baseline. Trouver ceux qui restent méthylés après traitement.
## Regardez uniquement les gènes, ou niqueement les CpGs puis gènes associés
```{r}
Response_diff_hypermet_in_Baseline <- BMIQ_wang_feng_analysis[["Baseline-Control"]] %>% 
  dplyr::filter(., P.Value < 0.01) %>%
  dplyr::filter(., logFC > 0.3)

Response_diff_hypermet_in_Control <- BMIQ_wang_feng_analysis[["Baseline-Control"]] %>% 
  dplyr::filter(., P.Value < 0.01) %>%
  dplyr::filter(., logFC < 0.3)


Genes_hypermetylated_in_Baseline <- Look_at_gene_with_CpGs_in_promoter(Response_diff_hypermet_in_Baseline$ID, match_hit_CpGs_Blueprint_promoter)

Genes_hypermetylated_in_Baseline_enhancer <- Look_at_genes_connected_to_CpGs(Response_diff_hypermet_in_Control$ID,
                                                                                  matchit_CpGs_Pchic, 
                                                                                  pchic,
                                                                                  matchit_Blueprint_Pchic,
                                                                                  match_hit_CpGs_Blueprint_promoter)
```

```{r}
Genes_hyper_in_Baseline_still_hyper_in_NR <- intersect(Genes_hypermetylated_in_Baseline,Genes_hypermetylated_in_Non_responder)

Genes_hyper_in_Baseline_still_hyper_in_NR_enhancer <- intersect(Genes_hypermetylated_in_Baseline_enhancer,Genes_hypermetylated_in_Non_responder_enhancer)
```
```{r}
Genes_hyper_in_Baseline_still_hyper_in_NR_ego <- enrichGO(
  gene = Genes_hyper_in_Baseline_still_hyper_in_NR,
  keyType = "SYMBOL",
  OrgDb = "org.Hs.eg.db",
  ont = "ALL",
  pAdjustMethod = "none"
)

Genes_hyper_in_Baseline_still_hyper_in_NR_enhancer_ego <- enrichGO(
  gene = Genes_hyper_in_Baseline_still_hyper_in_NR_enhancer,
  keyType = "SYMBOL",
  OrgDb = "org.Hs.eg.db",
  ont = "ALL",
  pAdjustMethod = "none"
)

dotplot(Genes_hyper_in_Baseline_still_hyper_in_NR_ego, showCategory = 30)
dotplot(Genes_hyper_in_Baseline_still_hyper_in_NR_enhancer_ego, showCategory = 30)

```




```{r}
Genes_hypermetylated_in_Non_responder_ego <- enrichGO(
  gene = Genes_hypermetylated_in_Non_responder,
  keyType = "SYMBOL",
  OrgDb = "org.Hs.eg.db",
  ont = "ALL",
  pAdjustMethod = "none"
)

dotplot(Genes_hypermetylated_in_Non_responder_ego, showCategory = 30)

```


```{r}
DATA_for_PCA <- t(as.data.frame(BMIQ_Wang_Feng_high_variance))
res.pca <- PCA(DATA_for_PCA, ncp = 5, graph = TRUE)

plotellipses(res.pca, 72553)
fviz_eig(res.pca, addlabels = TRUE)

```

```{r}
BMIQ_Whiele_high_variance <- Focus_high_variance_CpGs(BMIQ_Whiele, percentage = 10)

Focus_global_methylation(match_hit_CpGs_Blueprint, anno, Phenotype_Whiele, BMIQ_Whiele_high_variance, c("WT", "IDH2m"), fill_colour = c("#0000FF", "#777777", "#FF0000", "#00FF00"), Change_directory = "High_variance/")

Focus_global_methylation(match_hit_CpGs_Blueprint_promoter, anno_promoter, Phenotype_Whiele, BMIQ_Whiele_high_variance, c("WT", "IDH2m"), fill_colour = c("#0000FF", "#777777", "#FF0000", "#00FF00"), Change_directory = "High_variance/Promoter/")
```

