---
title: "R Notebook"
output: html_notebook
---

```{r}
setwd("~/Thesis/Projects/Epigenomic_integration/")
WD <- getwd()
source("Scripts/packages.R")
source("Scripts/functions.R")


```

```{r}
RGSet <- read.metharray.exp(paste0(WD, "/DATA/CD34/Methylation/"), recursive = T, force = TRUE)

# if input file is IlluminaHumanMethylation450k, change  #
#         outType = "IlluminaHumanMethylation450k"       #
#         outType = "IlluminaHumanMethylationEPIC"       #

RGSet <- convertArray(RGSet, outType = "IlluminaHumanMethylation450k")


Phenotype <- read.csv(paste0(WD, "/DATA/CD34/Methylation/Phenotype.csv"))
colnames(RGSet) <- Phenotype$Row.names

MSet <- preprocessRaw(RGSet)
qc <- getQC(MSet)
plotQC(qc)
meds <- (qc$uMed + qc$mMed) / 2
keepIndex <- which(meds > 10.5)
Good_samples <- colnames(RGSet)[which(index(colnames(RGSet)) %in% keepIndex)]

RawBeta <- champ.load(directory = paste0(WD,"/DATA/CD34/Methylation/"), arraytype = "450k")
champ.QC(beta = RawBeta$beta, pheno = RawBeta$pd$Row.names, resultsDir = "./Rawbeta_QC")


anno_450k <- read.csv("~/Illumina_manifest/HumanMethylation450_15017482_v1-2.csv", as.is = TRUE, skip = 7)
anno_450k <- anno_450k[, c("CHR", "MAPINFO", "Name", "UCSC_RefGene_Name")]
```

```{r}
BMIQ <- champ.norm(beta = RawBeta$beta, arraytype = "450K", cores = 12, resultsDir = "./BMIQ_Normalization/", method = "BMIQ")
```

```{r}
Phenotype_CD4 <- Phenotype %>% dplyr::filter(., .$DNMT3a.transcript == "dnmt3a transcript: Control")

BMIQ_CD4 <- BMIQ %>% .[,Phenotype_CD4$Row.names]

write.csv(BMIQ_CD4, file = paste0(WD, "/DATA/CD34/Methylation/BMIQ.csv"))

```

